--- a/src/dbus_fast/_private/address.py
+++ b/src/dbus_fast/_private/address.py
@@ -49,7 +49,7 @@ def get_system_bus_address() -> str:
     """Get the system bus address from the environment or return the default."""
     return (
         os.environ.get("DBUS_SYSTEM_BUS_ADDRESS")
-        or "unix:path=/var/run/dbus/system_bus_socket"
+        or "unix:path=/opt/var/run/dbus/system_bus_socket"
     )
 
 
@@ -81,7 +81,7 @@ def get_session_bus_address() -> str:
     # should be fairly reliable. fix this by passing in an async func to read
     # the file for each io backend.
     machine_id = None
-    with open("/var/lib/dbus/machine-id") as f:
+    with open("/opt/var/lib/dbus/machine-id") as f:
         machine_id = f.read().rstrip()
 
     dbus_info_file_name = f"{home}/.dbus/session-bus/{machine_id}-{display}"
