--- a/config/delegates.xml.in
+++ b/config/delegates.xml.in
@@ -62,31 +62,31 @@
   <delegate decode="bpg" command="&apos;@BPGDecodeDelegate@&apos; -b 16 -o &apos;%o&apos; &apos;%i&apos;"/>
   <delegate decode="png" encode="bpg" command="&apos;@BPGEncodeDelegate@&apos; -b 12 -q &apos;%~&apos; -o &apos;%o&apos; &apos;%i&apos;"/>
   <delegate decode="blender" command="&apos;@BlenderDecodeDelegate@&apos; -b &apos;%i&apos; -F PNG -o &apos;%o&apos;&apos;\n&apos;@ConvertDelegate@&apos; -concatenate &apos;%o*.png&apos; &apos;%o&apos;"/>
-  <delegate decode="browse" stealth="True" spawn="True" command="&apos;@BrowseDelegate@&apos; https://imagemagick.org/; @RMDelegate@ &apos;%i&apos;"/>
+  <delegate decode="browse" stealth="True" spawn="True" command="&apos;@BrowseDelegate@&apos; https://imagemagick.org/; /opt/bin/rm &apos;%i&apos;"/>
   <delegate decode="http:decode" command="&apos;@WWWDecodeDelegate@&apos; -s -L -o &apos;%u.dat&apos; &apos;http:%M&apos;"/>
   <delegate decode="https:decode" command="&apos;@WWWDecodeDelegate@&apos; -s -L -o &apos;%u.dat&apos; &apos;https:%M&apos;"/>
-  <delegate decode="doc" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="docx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="dng:decode" command="@MKDIRDelegate@ &apos;%u-config&apos;;@MVDelegate@ &apos;%i&apos; &apos;%i.%e&apos;; &apos;@DNGDecodeDelegate@&apos; --import &apos;%i.%e&apos; &apos;%u.tif&apos; --core --configdir &apos;%u-config&apos; &gt; &apos;%u&apos;;@RMDelegate@ -r &apos;%u-config&apos;"/>
+  <delegate decode="doc" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="docx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="dng:decode" command="@MKDIRDelegate@ &apos;%u-config&apos;;/opt/bin/mv &apos;%i&apos; &apos;%i.%e&apos;; &apos;@DNGDecodeDelegate@&apos; --import &apos;%i.%e&apos; &apos;%u.tif&apos; --core --configdir &apos;%u-config&apos; &gt; &apos;%u&apos;;/opt/bin/rm -r &apos;%u-config&apos;"/>
   <delegate decode="dot" command="&apos;@GVCDecodeDelegate@&apos; -Tsvg &apos;%i&apos; -o &apos;%o&apos;" />
   <delegate decode="dvi" command="&apos;@DVIDecodeDelegate@&apos; -sstdout=%%stderr -o &apos;%o&apos; &apos;%i&apos;"/>
   <delegate decode="edit" stealth="True" command="&apos;@EditorDelegate@&apos; -title &apos;Edit Image Comment&apos; -e vi &apos;%o&apos;"/>
-  <delegate decode="emf" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="emf" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
   <delegate decode="eps" encode="pdf" mode="bi" command="&apos;@PSDelegate@&apos; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 &apos;-sDEVICE=@GSPDFDevice@&apos; &apos;-sOutputFile=%o&apos; &apos;-f%i&apos;"/>
   <delegate decode="eps" encode="ps" mode="bi" command="&apos;@PSDelegate@&apos; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@GSPSDevice@&apos; &apos;-sOutputFile=%o&apos; &apos;-f%i&apos;"/>
-  <delegate decode="epub" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="hpg" command="&apos;@HPGLDecodeDelegate@&apos; -sstdout=%%stderr -m eps -f `basename &apos;%o&apos;` &apos;%i&apos;;     @MVDelegate@ -f `basename &apos;%o&apos;` &apos;%o&apos;"/>
-  <delegate decode="hpgl" command="&apos;@HPGLDecodeDelegate@&apos; -sstdout=%%stderr -m eps -f `basename &apos;%o&apos;` &apos;%i&apos;;     @MVDelegate@ -f `basename &apos;%o&apos;` &apos;%o&apos;"/>
+  <delegate decode="epub" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="hpg" command="&apos;@HPGLDecodeDelegate@&apos; -sstdout=%%stderr -m eps -f `basename &apos;%o&apos;` &apos;%i&apos;;     /opt/bin/mv -f `basename &apos;%o&apos;` &apos;%o&apos;"/>
+  <delegate decode="hpgl" command="&apos;@HPGLDecodeDelegate@&apos; -sstdout=%%stderr -m eps -f `basename &apos;%o&apos;` &apos;%i&apos;;     /opt/bin/mv -f `basename &apos;%o&apos;` &apos;%o&apos;"/>
   <delegate decode="htm" command="&apos;@HTMLDecodeDelegate@&apos; -U -o &apos;%o&apos; &apos;%i&apos;"/>
   <delegate decode="html" command="&apos;@HTMLDecodeDelegate@&apos; -U -o &apos;%o&apos; &apos;%i&apos;"/>
   <delegate decode="ilbm" command="&apos;@ILBMDecodeDelegate@&apos; &apos;%i&apos; &gt; &apos;%o&apos;"/>
   <delegate decode="jpg" encode="lep" mode="encode" command="&apos;@LEPDelegate@&apos; &apos;%i&apos; &apos;%o&apos;"/>
-  <delegate decode="jxr" command="@MVDelegate@ &apos;%i&apos; &apos;%i.jxr&apos;; &apos;@JXRDecodeDelegate@&apos; -i &apos;%i.jxr&apos; -o &apos;%o.tiff&apos;; @MVDelegate@ &apos;%i.jxr&apos; &apos;%i&apos;; @MVDelegate@ &apos;%o.tiff&apos; &apos;%o&apos;"/>
+  <delegate decode="jxr" command="/opt/bin/mv &apos;%i&apos; &apos;%i.jxr&apos;; &apos;@JXRDecodeDelegate@&apos; -i &apos;%i.jxr&apos; -o &apos;%o.tiff&apos;; /opt/bin/mv &apos;%i.jxr&apos; &apos;%i&apos;; /opt/bin/mv &apos;%o.tiff&apos; &apos;%o&apos;"/>
   <delegate decode="lep" mode="decode" command="&apos;@LEPDelegate@&apos; &apos;%i&apos; &apos;%o&apos;"/>
   <delegate decode="miff" encode="show" spawn="True" command="&apos;@DisplayDelegate@&apos; display -immutable -delay 0 -title &apos;%M&apos; &apos;%i&apos;"/>
   <delegate decode="miff" encode="win" stealth="True" spawn="True" command="&apos;@DisplayDelegate@&apos; display -immutable -delay 0 -title &apos;%M&apos; &apos;%i&apos;"/>
-  <delegate decode="odg" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="odt" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="odg" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="odt" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
   <delegate decode="pcl:cmyk" stealth="True" command="&apos;@PCLDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@PCLCMYKDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
   <delegate decode="pcl:color" stealth="True" command="&apos;@PCLDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@PCLColorDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
   <delegate decode="pcl:mono" stealth="True" command="&apos;@PCLDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@PCLMonoDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
@@ -97,10 +97,10 @@
   <delegate decode="clipboard" command="&apos;@ClipboardDelegate@&apos; -selection clipboard -o &gt; &apos;%o&apos;"/>
   <delegate decode="png" encode="webp" command="&apos;@WebPEncodeDelegate@&apos; -quiet -q %Q &apos;%i&apos; -o &apos;%o&apos;"/>
   <delegate decode="pnm" encode="ilbm" mode="encode" command="&apos;@ILBMEncodeDelegate@&apos; -24if &apos;%i&apos; &gt; &apos;%o&apos;"/>
-  <delegate decode="tiff" encode="jxr" command="@MVDelegate@ &apos;%i&apos; &apos;%i.tiff&apos;; &apos;@JXREncodeDelegate@&apos; -i &apos;%i.tiff&apos; -o &apos;%o.jxr&apos; -q %Q; @MVDelegate@ &apos;%i.tiff&apos; &apos;%i&apos;; @MVDelegate@ &apos;%o.jxr&apos; &apos;%o&apos;"/>
-  <delegate decode="tiff" encode="wdp" command="@MVDelegate@ &apos;%i&apos; &apos;%i.tiff&apos;; &apos;@JXREncodeDelegate@&apos; -i &apos;%i.tiff&apos; -o &apos;%o.jxr&apos; -q %Q; @MVDelegate@ &apos;%i.tiff&apos; &apos;%i&apos;; @MVDelegate@ &apos;%o.jxr&apos; &apos;%o&apos;"/>
-  <delegate decode="ppt" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="pptx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="tiff" encode="jxr" command="/opt/bin/mv &apos;%i&apos; &apos;%i.tiff&apos;; &apos;@JXREncodeDelegate@&apos; -i &apos;%i.tiff&apos; -o &apos;%o.jxr&apos; -q %Q; /opt/bin/mv &apos;%i.tiff&apos; &apos;%i&apos;; /opt/bin/mv &apos;%o.jxr&apos; &apos;%o&apos;"/>
+  <delegate decode="tiff" encode="wdp" command="/opt/bin/mv &apos;%i&apos; &apos;%i.tiff&apos;; &apos;@JXREncodeDelegate@&apos; -i &apos;%i.tiff&apos; -o &apos;%o.jxr&apos; -q %Q; /opt/bin/mv &apos;%i.tiff&apos; &apos;%i&apos;; /opt/bin/mv &apos;%o.jxr&apos; &apos;%o&apos;"/>
+  <delegate decode="ppt" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="pptx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
   <delegate decode="ps" encode="prt" command="&apos;@LPRDelegate@&apos; &apos;%i&apos;"/>
   <delegate decode="ps:alpha" stealth="True" command="&apos;@PSDelegate@&apos; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@GSAlphaDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;-f%s&apos; &apos;-f%s&apos;"/>
   <delegate decode="ps:cmyk" stealth="True" command="&apos;@PSDelegate@&apos; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@GSCMYKDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;-f%s&apos; &apos;-f%s&apos;"/>
@@ -114,18 +114,18 @@
   <delegate decode="svg" command="&apos;@RSVGDecodeDelegate@&apos; --dpi-x %x --dpi-y %y -o &apos;%o&apos; &apos;%i&apos;"/>
   <!-- Change export-filename to export-png for inkscape < 1.0 -->
   <delegate decode="svg:decode" stealth="True" command="&apos;@SVGDecodeDelegate@&apos; &apos;%s&apos; &apos;--export-filename=%s&apos; &apos;--export-dpi=%s&apos; &apos;--export-background=%s&apos; &apos;--export-background-opacity=%s&apos;"/>
-  <delegate decode="tiff" encode="text" command="&apos;@TextEncodeDelegate@&apos; &apos;%i&apos; &apos;%u&apos;; @MVDelegate@ &apos;%u.txt&apos; &apos;%o&apos;"/>
+  <delegate decode="tiff" encode="text" command="&apos;@TextEncodeDelegate@&apos; &apos;%i&apos; &apos;%u&apos;; /opt/bin/mv &apos;%u.txt&apos; &apos;%o&apos;"/>
   <delegate decode="tiff" encode="launch" mode="encode" command="&apos;@LaunchDelegate@&apos; &apos;%i&apos;"/>
-  <delegate decode="wdp" command="@MVDelegate@ &apos;%i&apos; &apos;%i.jxr&apos;; &apos;@JXRDecodeDelegate@&apos; -i &apos;%i.jxr&apos; -o &apos;%o.tiff&apos;; @MVDelegate@ &apos;%i.jxr&apos; &apos;%i&apos;; @MVDelegate@ &apos;%o.tiff&apos; &apos;%o&apos;"/>
+  <delegate decode="wdp" command="/opt/bin/mv &apos;%i&apos; &apos;%i.jxr&apos;; &apos;@JXRDecodeDelegate@&apos; -i &apos;%i.jxr&apos; -o &apos;%o.tiff&apos;; /opt/bin/mv &apos;%i.jxr&apos; &apos;%i&apos;; /opt/bin/mv &apos;%o.tiff&apos; &apos;%o&apos;"/>
   <delegate decode="webp" command="&apos;@WebPDecodeDelegate@&apos; -pam &apos;%i&apos; -o &apos;%o&apos;"/>
-  <delegate decode="wmf" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="xls" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="xlsx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="wmf" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="xls" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="xlsx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
   <delegate decode="xps:cmyk" stealth="True" command="&apos;@XPSDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@XPSCMYKDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
   <delegate decode="xps:color" stealth="True" command="&apos;@XPSDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@XPSColorDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
   <delegate decode="xps:mono" stealth="True" command="&apos;@XPSDelegate@&apos; -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 &apos;-sDEVICE=@XPSMonoDevice@&apos; -dTextAlphaBits=%u -dGraphicsAlphaBits=%u &apos;-r%s&apos; %s &apos;-sOutputFile=%s&apos; &apos;%s&apos;"/>
   <delegate decode="video:decode" command="&apos;@VIDEODecodeDelegate@&apos; -nostdin -loglevel error -i &apos;%s&apos; -an -f rawvideo -y %s &apos;%s&apos;"/>
   <delegate encode="video:encode" stealth="True" command="&apos;@VIDEOEncodeDelegate@&apos; -nostdin -loglevel error -i &apos;%s%%d.%s&apos; %s &apos;%s.%s&apos;"/>
-  <delegate decode="vsd" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
-  <delegate decode="vsdx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; @MVDelegate@ &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="vsd" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
+  <delegate decode="vsdx" command="&apos;@DOCDecodeDelegate@&apos; --convert-to pdf --outdir `dirname &apos;%i&apos;` &apos;%i&apos; 2&gt; &apos;%u&apos;; /opt/bin/mv &apos;%i.pdf&apos; &apos;%o&apos;"/>
 </delegatemap>
--- a/config/mime.xml
+++ b/config/mime.xml
@@ -213,16 +213,8 @@
   <mime type="application/x-arj" description="ARJ archive" data-type="short" endian="LSB" offset="0" magic="0xea60" priority="50" />
   <mime type="application/x-arj" acronym="ARJ" description="ARJ archive" priority="100" pattern="*.arj" />
   <mime type="application/x-asp" acronym="ASP" description="Active Server Page" priority="100" pattern="*.asp" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#! /bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/usr/bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#! /usr/bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/usr/local/bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#! /usr/local/bin/gawk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/bin/awk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#! /bin/awk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/usr/bin/awk" priority="50" />
-  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#! /usr/bin/awk" priority="50" />
+  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/opt/bin/gawk" priority="50" />
+  <mime type="application/x-awk" description="AWK script" data-type="string" offset="0" magic="#!/opt/bin/awk" priority="50" />
   <mime type="application/x-bcpio" acronym="BCPIO" description="BCPIO document" priority="100" pattern="*.bcpio" />
   <mime type="application/x-bittorrent" description="BitTorrent seed file" data-type="string" offset="0" magic="d8:announce" priority="50" />
   <mime type="application/x-bittorrent" description="BitTorrent seed file" priority="100" pattern="*.torrent" />
@@ -264,10 +256,10 @@
   <mime type="application/x-cpio" description="CPIO archive" data-type="short" offset="0" magic="0143561" priority="50" />
   <mime type="application/x-cpio" description="CPIO archive" priority="100" pattern="*.cpio" />
   <mime type="application/x-cpio-compressed" description="CPIO archive (gzip-compressed)" priority="100" pattern="*.cpio.gz" />
-  <mime type="application/x-csh" description="C shell script" data-type="string" offset="1:16" magic="/bin/tcsh" />
-  <mime type="application/x-csh" description="C shell script" data-type="string" offset="1:16" magic="/bin/csh" />
-  <mime type="application/x-csh" description="C shell script" data-type="string" offset="0" magic="#!/usr/bin/env csh" />
-  <mime type="application/x-csh" description="C shell script" data-type="string" offset="0" magic="#!/usr/bin/env tcsh" />
+  <mime type="application/x-csh" description="C shell script" data-type="string" offset="1:16" magic="/opt/bin/tcsh" />
+  <mime type="application/x-csh" description="C shell script" data-type="string" offset="1:16" magic="/opt/bin/csh" />
+  <mime type="application/x-csh" description="C shell script" data-type="string" offset="0" magic="#!/opt/bin/csh" />
+  <mime type="application/x-csh" description="C shell script" data-type="string" offset="0" magic="#!/opt/bin/tcsh" />
   <mime type="application/x-csh" description="C shell script" priority="100" pattern="*.csh" />
   <mime type="application/x-dbf" description="Xbase document" priority="100" pattern="*.dbf" />
   <mime type="application/ecmascript" description="ECMAScript program" priority="100" pattern="*.es" />
@@ -494,9 +486,8 @@
   <mime type="application/x-par2" description="PAR2 Parity File" priority="100" pattern="*.PAR2" />
   <mime type="application/x-par2" description="PAR2 Parity File" priority="100" pattern="*.par2" />
   <mime type="application/x-pef-executable" description="PEF executable" data-type="string" offset="0" magic="Joy!" priority="50" />
-  <mime type="application/x-perl" description="Perl script" data-type="string" offset="0" magic="eval \&quot;exec /usr/local/bin/perl" priority="50" />
-  <mime type="application/x-perl" description="Perl script" data-type="string" offset="1:16" magic="/bin/perl" priority="50" />
-  <mime type="application/x-perl" description="Perl script" data-type="string" offset="1:16" magic="/bin/env perl" priority="50" />
+  <mime type="application/x-perl" description="Perl script" data-type="string" offset="0" magic="eval \&quot;exec /opt/bin/perl" priority="50" />
+  <mime type="application/x-perl" description="Perl script" data-type="string" offset="1:16" magic="/opt/bin/perl" priority="50" />
   <mime type="application/x-perl" description="Perl script" priority="100" pattern="*.pl" />
   <mime type="application/x-perl" description="Perl script" priority="100" pattern="*.pm" />
   <mime type="application/x-perl" description="Perl script" priority="100" pattern="*.al" />
@@ -529,8 +520,7 @@
   <mime type="application/x-reject" description="rejected patch" priority="100" pattern="*.rej" />
   <mime type="application/x-rpm" description="RPM package" data-type="string" offset="0" magic="\xed\xab\xee\xdb" priority="50" />
   <mime type="application/x-rpm" description="RPM package" priority="100" pattern="*.rpm" />
-  <mime type="application/x-ruby" description="Ruby script" data-type="string" offset="1:16" magic="/bin/env ruby" />
-  <mime type="application/x-ruby" description="Ruby script" data-type="string" offset="1:16" magic="/bin/ruby" />
+  <mime type="application/x-ruby" description="Ruby script" data-type="string" offset="1:16" magic="/opt/bin/ruby" />
   <mime type="application/x-ruby" description="Ruby script" priority="100" pattern="*.rb" />
   <mime type="application/x-markaby" description="Markaby script" priority="100" pattern="*.mab" />
   <mime type="application/x-sc" description="SC/Xspread spreadsheet" data-type="string" offset="38" magic="Spreadsheet" priority="50" />
@@ -542,15 +532,15 @@
   <mime type="application/x-sharedlib" description="shared library" data-type="string" offset="0" mask="0xffffffff000000000000000000000000ff" magic="\177ELF \003" priority="50" />
   <mime type="application/x-sharedlib" description="shared library" priority="100" pattern="*.so" />
   <mime type="application/x-shellscript" description="shell script" data-type="string" offset="10" magic="# This is a shell archive" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/bin/bash" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/bin/nawk" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/bin/zsh" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/bin/sh" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/bin/ksh" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/usr/bin/env sh" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/usr/bin/env bash" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/usr/bin/env zsh" priority="50" />
-  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/usr/bin/env ksh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/opt/bin/bash" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/opt/bin/nawk" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/opt/bin/zsh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/opt/bin/sh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="1:16" magic="/opt/bin/ksh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/opt/bin/sh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/opt/bin/bash" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/opt/bin/zsh" priority="50" />
+  <mime type="application/x-shellscript" description="shell script" data-type="string" offset="0" magic="#!/opt/bin/ksh" priority="50" />
   <mime type="application/x-shellscript" description="shell script" priority="100" pattern="*.sh" />
   <mime type="application/x-shockwave-flash" description="Shockwave Flash file" data-type="string" offset="0" magic="FWS" priority="50" />
   <mime type="application/x-shockwave-flash" description="Shockwave Flash file" data-type="string" offset="0" magic="CWS" priority="50" />
@@ -1014,8 +1004,7 @@
   <mime type="text/x-ldif" acronym="LDIF" description="LDAP Data Interchange Format" priority="100" pattern="*.ldif" />
   <mime type="text/x-literate-haskell" description="Literate Haskell source code" priority="100" pattern="*.lhs" />
   <mime type="text/x-log" description="application log" priority="100" pattern="*.log" />
-  <mime type="text/x-makefile" description="Makefile" data-type="string" offset="0" magic="#!/usr/bin/make" priority="50" />
-  <mime type="text/x-makefile" description="Makefile" data-type="string" offset="0" magic="#! /usr/bin/make" priority="50" />
+  <mime type="text/x-makefile" description="Makefile" data-type="string" offset="0" magic="#!/opt/bin/make" priority="50" />
   <mime type="text/x-makefile" description="Makefile" priority="100" pattern="[Mm]akefile" />
   <mime type="text/x-makefile" description="Makefile" priority="100" pattern="GNUmakefile" />
   <mime type="text/x-moc" acronym="Qt MOC" description="Qt Meta Object Compiler" priority="100" pattern="*.moc" />
@@ -1042,19 +1031,11 @@
   <mime type="text/x-patch" description="differences between files" data-type="string" offset="0" magic="Index:" priority="50" />
   <mime type="text/x-patch" description="differences between files" priority="100" pattern="*.diff" />
   <mime type="text/x-patch" description="differences between files" priority="100" pattern="*.patch" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#!/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#! /bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="eval \&quot;exec /bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#!/usr/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#! /usr/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="eval \&quot;exec /usr/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#!/usr/local/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#! /usr/local/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="eval \&quot;exec /usr/local/bin/python" priority="50" />
-  <mime type="text/x-python" description="Python script" data-type="string" offset="1:16" magic="/bin/env python" priority="50" />
+  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="#!/opt/bin/python" priority="50" />
+  <mime type="text/x-python" description="Python script" data-type="string" offset="0" magic="eval \&quot;exec /opt/bin/python" priority="50" />
+  <mime type="text/x-python" description="Python script" data-type="string" offset="1:16" magic="/opt/bin/python" priority="50" />
   <mime type="text/x-python" description="Python script" priority="100" pattern="*.py" />
-  <mime type="text/x-lua" description="Lua script" data-type="string" offset="0" magic="/bin/lua" priority="50" />
-  <mime type="text/x-lua" description="Lua script" data-type="string" offset="0" magic="/bin/env lua" priority="50" />
+  <mime type="text/x-lua" description="Lua script" data-type="string" offset="0" magic="/opt/bin/lua" priority="50" />
   <mime type="text/x-lua" description="Lua script" priority="100" pattern="*.lua" />
   <mime type="text/x-readme" description="README document" priority="100" pattern="README*" />
   <mime type="text/x-readme" description="README document" priority="100" pattern="*.nfo" />
