--- a/alsactl/alsactl.c
+++ b/alsactl/alsactl.c
@@ -31,18 +31,14 @@
 #include <sched.h>
 #include "alsactl.h"
 
-#ifndef SYS_ASOUND_DIR
-#define SYS_ASOUND_DIR "/var/lib/alsa"
-#endif
-#ifndef SYS_ASOUNDRC
+#undef SYS_ASOUND_DIR
+#define SYS_ASOUND_DIR "/opt/var/lib/alsa"
+#undef SYS_ASOUNDRC
 #define SYS_ASOUNDRC SYS_ASOUND_DIR "/asound.state"
-#endif
-#ifndef SYS_PIDFILE
-#define SYS_PIDFILE "/var/run/alsactl.pid"
-#endif
-#ifndef SYS_LOCKPATH
-#define SYS_LOCKPATH "/var/lock"
-#endif
+#undef SYS_PIDFILE
+#define SYS_PIDFILE "/opt/var/run/alsactl.pid"
+#undef SYS_LOCKPATH
+#define SYS_LOCKPATH "/opt/var/lock"
 
 int debugflag = 0;
 int force_restore = 1;
