--- a/configure.in
+++ b/configure.in
@@ -653,22 +653,22 @@ if test "$class_driver" = "no" && test "
    AC_CHECK_PROGS(PYTHON,[python python3 python2])
    AM_PATH_PYTHON([2.2])
    AC_MSG_CHECKING([for path to Python.h])
-   PYTHONINCLUDEDIR=`$PYTHON -c "from distutils.sysconfig import get_python_inc; print (get_python_inc());" 2>/dev/null || echo "not found"`
-   if test "$PYTHONINCLUDEDIR" = "not found"; then
-      AC_MSG_WARN([Failed to determine the Python include directory. Retry using sysconfig module.])
-      PYTHONINCLUDEDIR=`$PYTHON -c "import sysconfig; print(sysconfig.get_path('include'))" 2>/dev/null || echo "not found"`
-      if test "$PYTHONINCLUDEDIR" = "not found"; then
-         AC_MSG_ERROR([Failed to determine the Python include directory. Please check your Python installation.], 6)
-      fi
-   fi
-   PYTHONEXECDIR=`$PYTHON -c "from distutils.sysconfig import get_python_lib; print (get_python_lib(1));" 2>/dev/null || echo "not found"`
-   if test "$PYTHONEXECDIR" = "not found"; then
-      AC_MSG_WARN([Failed to determine the Python executable directory. Retry using sysconfig module.])
-      PYTHONEXECDIR=`$PYTHON -c "import sysconfig; print(sysconfig.get_path('platlib'))" 2>/dev/null || echo "not found"`
-      if test "$PYTHONEXECDIR" = "not found"; then
-         AC_MSG_ERROR([Failed to determine the Python executable directory. Please check your Python installation.], 6)
-      fi
-   fi
+   #PYTHONINCLUDEDIR=`$PYTHON -c "from distutils.sysconfig import get_python_inc; print (get_python_inc());" 2>/dev/null || echo "not found"`
+   #if test "$PYTHONINCLUDEDIR" = "not found"; then
+   #   AC_MSG_WARN([Failed to determine the Python include directory. Retry using sysconfig module.])
+   #   PYTHONINCLUDEDIR=`$PYTHON -c "import sysconfig; print(sysconfig.get_path('include'))" 2>/dev/null || echo "not found"`
+   #   if test "$PYTHONINCLUDEDIR" = "not found"; then
+   #      AC_MSG_ERROR([Failed to determine the Python include directory. Please check your Python installation.], 6)
+   #   fi
+   #fi
+   #PYTHONEXECDIR=`$PYTHON -c "from distutils.sysconfig import get_python_lib; print (get_python_lib(1));" 2>/dev/null || echo "not found"`
+   #if test "$PYTHONEXECDIR" = "not found"; then
+   #   AC_MSG_WARN([Failed to determine the Python executable directory. Retry using sysconfig module.])
+   #   PYTHONEXECDIR=`$PYTHON -c "import sysconfig; print(sysconfig.get_path('platlib'))" 2>/dev/null || echo "not found"`
+   #   if test "$PYTHONEXECDIR" = "not found"; then
+   #      AC_MSG_ERROR([Failed to determine the Python executable directory. Please check your Python installation.], 6)
+   #   fi
+   #fi
    AC_MSG_RESULT("using python executable directory $PYTHONEXECDIR ....  python${PYTHON_VERSION}/Python.h")
    AC_MSG_RESULT("using python include directory $PYTHONINCLUDEDIR ....  python${PYTHON_VERSION}/Python.h")
    AC_ARG_VAR(PYTHONINCLUDEDIR, [path to Python.h C header file])
