# https://github.com/DanBloomberg/leptonica/commit/e0936377f9094b72f7f2ea6bc34311ccc69b2f71

--- a/cmake/LeptonicaFunc.cmake
+++ b/cmake/LeptonicaFunc.cmake
@@ -3,32 +3,32 @@
 # ##############################################################################
 # FUNCTION find_and_handle_library
 # ##############################################################################
-function(find_and_handle_library LIB_NAME ENABLE_OPTION PKG_CONFIG_NAME)
+function(find_and_handle_library PACKAGE_NAME ENABLE_OPTION PKG_CONFIG_NAME)
   if(${ENABLE_OPTION})
-    find_package(${PKG_CONFIG_NAME})
-    if(${PKG_CONFIG_NAME}_FOUND)
+    find_package(${PACKAGE_NAME})
+    if(${PACKAGE_NAME}_FOUND)
       set(pkgs_private
           "${pkgs_private} ${PKG_CONFIG_NAME}"
           PARENT_SCOPE)
       # Propagate found variables to parent scope
-      set(${PKG_CONFIG_NAME}_FOUND
-          ${${PKG_CONFIG_NAME}_FOUND}
+      set(${PACKAGE_NAME}_FOUND
+          ${${PACKAGE_NAME}_FOUND}
           PARENT_SCOPE)
-      if(DEFINED ${PKG_CONFIG_NAME}_LIBRARIES)
-        set(${PKG_CONFIG_NAME}_LIBRARIES
-            ${${PKG_CONFIG_NAME}_LIBRARIES}
+      if(DEFINED ${PACKAGE_NAME}_LIBRARIES)
+        set(${PACKAGE_NAME}_LIBRARIES
+            ${${PACKAGE_NAME}_LIBRARIES}
             PARENT_SCOPE)
       endif()
-      if(DEFINED ${PKG_CONFIG_NAME}_INCLUDE_DIRS)
-        set(${PKG_CONFIG_NAME}_INCLUDE_DIRS
-            ${${PKG_CONFIG_NAME}_INCLUDE_DIRS}
+      if(DEFINED ${PACKAGE_NAME}_INCLUDE_DIRS)
+        set(${PACKAGE_NAME}_INCLUDE_DIRS
+            ${${PACKAGE_NAME}_INCLUDE_DIRS}
             PARENT_SCOPE)
       endif()
     else()
       if(STRICT_CONF)
         message(
           FATAL_ERROR
-            "Could not find ${PKG_CONFIG_NAME} libs. Use -D${ENABLE_OPTION}=OFF to disable ${PKG_CONFIG_NAME} support."
+          "Could not find ${PACKAGE_NAME} libs. Use -D${ENABLE_OPTION}=OFF to disable ${PACKAGE_NAME} support."
         )
       endif()
     endif()
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,10 +81,10 @@ if(NOT SW_BUILD)
   set(pkgs_private "")
   set(libs_private "")
 
-  find_and_handle_library(ZLIB ENABLE_ZLIB ZLIB)
-  find_and_handle_library(PNG ENABLE_PNG PNG)
-  find_and_handle_library(JPEG ENABLE_JPEG JPEG)
-  find_and_handle_library(TIFF ENABLE_TIFF TIFF)
+  find_and_handle_library(ZLIB ENABLE_ZLIB zlib)
+  find_and_handle_library(PNG ENABLE_PNG libpng)
+  find_and_handle_library(JPEG ENABLE_JPEG libjpeg)
+  find_and_handle_library(TIFF ENABLE_TIFF libtiff-4)
 
   if(ENABLE_GIF)
     find_package(GIF 5)
