--- a/include/libnetfilter_log/libnetfilter_log.h
+++ b/include/libnetfilter_log/libnetfilter_log.h
@@ -9,8 +9,15 @@
 #ifndef __LIBNETFILTER_LOG_H
 #define __LIBNETFILTER_LOG_H
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
 #include <stdint.h>
 #include <sys/types.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,37)
+#include <sys/socket.h> // fix sa_family_t /* AF_NETLINK */
+#endif
 #include <linux/netlink.h>
 #include <libnetfilter_log/linux_nfnetlink_log.h>
 
