--- a/lib/param/loadparm.c
+++ b/lib/param/loadparm.c
@@ -2683,7 +2683,7 @@ struct loadparm_context *loadparm_init(T
 				      MIT_KDC_PATH);
 #endif
 	lpcfg_do_global_parameter(lp_ctx, "template shell", "/bin/false");
-	lpcfg_do_global_parameter(lp_ctx, "template homedir", "/home/%D/%U");
+	lpcfg_do_global_parameter(lp_ctx, "template homedir", "/opt/home/%D/%U");
 
 	lpcfg_do_global_parameter(lp_ctx, "client signing", "default");
 	lpcfg_do_global_parameter(lp_ctx, "client ipc signing", "default");
--- a/source3/param/loadparm.c
+++ b/source3/param/loadparm.c
@@ -796,7 +796,7 @@ static void init_globals(struct loadparm
 
 	lpcfg_string_set(Globals.ctx, &Globals.template_shell, "/bin/false");
 	lpcfg_string_set(Globals.ctx, &Globals.template_homedir,
-			 "/home/%D/%U");
+			 "/opt/home/%D/%U");
 	lpcfg_string_set(Globals.ctx, &Globals.winbind_separator, "\\");
 	lpcfg_string_set(Globals.ctx, &Globals.winbindd_socket_directory,
 			 dyn_WINBINDD_SOCKET_DIR);
--- a/source3/smbd/notifyd/test_notifyd.c
+++ b/source3/smbd/notifyd/test_notifyd.c
@@ -218,10 +218,10 @@ static bool test_notifyd_trigger1(struct
 	status = fcn_test(
 		msg_ctx,
 		notifyd,
-		"/home",
+		"/opt/home",
 		UINT32_MAX,
 		UINT32_MAX,
-		"/home/foo",
+		"/opt/home/foo",
 		UINT32_MAX,
 		UINT32_MAX,
 		&got_trigger);
