--- a/utils/mbim-network.in
+++ b/utils/mbim-network.in
@@ -37,7 +37,7 @@ help ()
     echo "   2) The mbim-network script requires a profile to work. Unless"
     echo "   explicitly specified with \`--profile', the file is assumed to"
     echo "   be available in the following path:"
-    echo "      /etc/mbim-network.conf"
+    echo "      /opt/etc/mbim-network.conf"
     echo
     echo "   3) The APN to use should be configured in the profile, in the"
     echo "   following way (e.g. assuming APN is called 'internet'):"
@@ -90,7 +90,7 @@ if [ $# -lt 2 ]; then
 fi
 
 # Defaults
-PROFILE_FILE=/etc/mbim-network.conf
+PROFILE_FILE=/opt/etc/mbim-network.conf
 
 # Device + Command with options; options given first
 while [ $# -gt 2 ]; do
