--- a/include/libssh/libcrypto.h
+++ b/include/libssh/libcrypto.h
@@ -126,7 +126,9 @@ int pki_key_ecgroup_name_to_nid(const ch
 #if defined(WITH_PKCS11_PROVIDER)
 int pki_load_pkcs11_provider(void);
 #else
+#ifndef OPENSSL_NO_ENGINE
 ENGINE *pki_get_engine(void);
+#endif /* OPENSSL_NO_ENGINE */
 #endif
 #endif /* WITH_PKCS11_PROVIDER */
 
--- a/src/libcrypto.c
+++ b/src/libcrypto.c
@@ -54,7 +54,7 @@
 #include <openssl/provider.h>
 #endif /* OPENSSL_VERSION_NUMBER */
 #include <openssl/rand.h>
-#if defined(WITH_PKCS11_URI) && !defined(WITH_PKCS11_PROVIDER)
+#if defined(WITH_PKCS11_URI) && !defined(WITH_PKCS11_PROVIDER) && !defined(OPENSSL_NO_ENGINE)
 #include <openssl/engine.h>
 #endif
 
@@ -98,7 +98,7 @@ void ssh_reseed(void){
 }
 
 #if defined(WITH_PKCS11_URI)
-#if defined(WITH_PKCS11_PROVIDER)
+#if defined(WITH_PKCS11_PROVIDER) && !defined(OPENSSL_NO_ENGINE)
 static OSSL_PROVIDER *provider = NULL;
 static bool pkcs11_provider_failed = false;
 
--- a/src/pki_crypto.c
+++ b/src/pki_crypto.c
@@ -33,7 +33,7 @@
 
 #include <openssl/pem.h>
 #include <openssl/evp.h>
-#if defined(WITH_PKCS11_URI) && !defined(WITH_PKCS11_PROVIDER)
+#if defined(WITH_PKCS11_URI) && !defined(WITH_PKCS11_PROVIDER) && !defined(OPENSSL_NO_ENGINE)
 #include <openssl/engine.h>
 #endif
 #include <openssl/err.h>
@@ -2744,7 +2744,7 @@ int pki_uri_import(const char *uri_name,
 #if OPENSSL_VERSION_NUMBER < 0x30000000L && HAVE_OPENSSL_ECC
     EC_KEY *ecdsa = NULL;
 #endif
-#ifndef WITH_PKCS11_PROVIDER
+#if !defined(WITH_PKCS11_PROVIDER) && !defined(OPENSSL_NO_ENGINE)
     ENGINE *engine = NULL;
 
     /* Do the init only once */
