--- a/mkfs/main.c
+++ b/mkfs/main.c
@@ -53,6 +53,10 @@
 #include "common/units.h"
 #include "check/qgroup-verify.h"
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
 struct mkfs_allocation {
 	u64 data;
 	u64 metadata;
@@ -1349,11 +1353,14 @@ int BOX_MAIN(mkfs)(int argc, char **argv
 	if (ret)
 		goto error;
 
+/* <linux/blkzoned.h> since Linux 4.10 */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
 	if (zoned && (!zoned_profile_supported(metadata_profile) ||
 		      !zoned_profile_supported(data_profile))) {
 		error("zoned mode does not yet support RAID/DUP profiles, please specify '-d single -m single' manually");
 		goto error;
 	}
+#endif /* <linux/blkzoned.h> since Linux 4.10 */
 
 	dev_cnt--;
 
