--- a/src/config.c
+++ b/src/config.c
@@ -56,8 +56,8 @@
 
 #include <getopt.h>
 
-#define OLD_DEFAULT_CFG_FILE "/etc/ocserv.conf"
-#define DEFAULT_CFG_FILE "/etc/ocserv/ocserv.conf"
+#define OLD_DEFAULT_CFG_FILE "/opt/etc/ocserv.conf"
+#define DEFAULT_CFG_FILE "/opt/etc/ocserv/ocserv.conf"
 
 static void print_version(void);
 
--- a/src/main-user.c
+++ b/src/main-user.c
@@ -47,7 +47,7 @@
 #include <script-list.h>
 #include <ccan/list/list.h>
 
-#define OCSERV_FW_SCRIPT "/usr/bin/ocserv-fw"
+#define OCSERV_FW_SCRIPT "/opt/bin/ocserv-fw"
 
 #define APPEND_TO_STR(str, val) \
 			ret = str_append_str(str, val); \
--- a/src/occtl/ctl.h
+++ b/src/occtl/ctl.h
@@ -1,7 +1,7 @@
 #ifndef CTL_H
 # define CTL_H
 
-#define OCCTL_UNIX_SOCKET "/var/run/occtl.socket"
+#define OCCTL_UNIX_SOCKET "/opt/var/run/occtl.socket"
 
 enum {
 	CTL_CMD_STATUS = 1,
--- a/src/ocpasswd/ocpasswd.c
+++ b/src/ocpasswd/ocpasswd.c
@@ -42,7 +42,7 @@
 #include <getpass.h>
 #include <minmax.h>
 
-#define DEFAULT_OCPASSWD "/etc/ocserv/ocpasswd"
+#define DEFAULT_OCPASSWD "/opt/etc/ocserv/ocpasswd"
 
 static const char alphabet[] =
 	"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ./";
