--- a/lib/mount_util.c
+++ b/lib/mount_util.c
@@ -117,9 +117,9 @@ static int add_mount(const char *prognam
 			goto out_restore;
 		}
 
-		execle("/bin/mount", "/bin/mount", "--no-canonicalize", "-i",
+		execle("/opt/bin/mount", "/opt/bin/mount", "--no-canonicalize", "-i",
 		       "-f", "-t", type, "-o", opts, fsname, mnt, NULL, &env);
-		fprintf(stderr, "%s: failed to execute /bin/mount: %s\n",
+		fprintf(stderr, "%s: failed to execute /opt/bin/mount: %s\n",
 			progname, strerror(errno));
 		exit(1);
 	}
@@ -177,13 +177,13 @@ static int exec_umount(const char *progn
 		}
 
 		if (lazy) {
-			execle("/bin/umount", "/bin/umount", "-i", rel_mnt,
+			execle("/opt/bin/umount", "/opt/bin/umount", "-i", rel_mnt,
 			       "-l", NULL, &env);
 		} else {
-			execle("/bin/umount", "/bin/umount", "-i", rel_mnt,
+			execle("/opt/bin/umount", "/opt/bin/umount", "-i", rel_mnt,
 			       NULL, &env);
 		}
-		fprintf(stderr, "%s: failed to execute /bin/umount: %s\n",
+		fprintf(stderr, "%s: failed to execute /opt/bin/umount: %s\n",
 			progname, strerror(errno));
 		exit(1);
 	}
@@ -248,9 +248,9 @@ static int remove_mount(const char *prog
 			goto out_restore;
 		}
 
-		execle("/bin/umount", "/bin/umount", "--no-canonicalize", "-i",
+		execle("/opt/bin/umount", "/opt/bin/umount", "--no-canonicalize", "-i",
 		       "--fake", mnt, NULL, &env);
-		fprintf(stderr, "%s: failed to execute /bin/umount: %s\n",
+		fprintf(stderr, "%s: failed to execute /opt/bin/umount: %s\n",
 			progname, strerror(errno));
 		exit(1);
 	}
