--- a/configure
+++ b/configure
@@ -8629,8 +8629,8 @@
     if test "x$enable_utf8" != xno; then
 	OLDLIBS="$LIBS"
 	if test -n "$ac_tool_prefix"; then
-  # Extract the first word of "${ac_tool_prefix}ncursesw5-config", so it can be a program name with args.
-set dummy ${ac_tool_prefix}ncursesw5-config; ac_word=$2
+  # Extract the first word of "${ac_tool_prefix}ncursesw6-config", so it can be a program name with args.
+set dummy ${ac_tool_prefix}ncursesw6-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_prog_NCURSESW_CONFIG+:} false; then :
@@ -8646,7 +8646,7 @@
   test -z "$as_dir" && as_dir=.
     for ac_exec_ext in '' $ac_executable_extensions; do
   if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_NCURSESW_CONFIG="${ac_tool_prefix}ncursesw5-config"
+    ac_cv_prog_NCURSESW_CONFIG="${ac_tool_prefix}ncursesw6-config"
     $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
     break 2
   fi
@@ -8669,8 +8669,8 @@
 fi
 if test -z "$ac_cv_prog_NCURSESW_CONFIG"; then
   ac_ct_NCURSESW_CONFIG=$NCURSESW_CONFIG
-  # Extract the first word of "ncursesw5-config", so it can be a program name with args.
-set dummy ncursesw5-config; ac_word=$2
+  # Extract the first word of "ncursesw6-config", so it can be a program name with args.
+set dummy ncursesw6-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_prog_ac_ct_NCURSESW_CONFIG+:} false; then :
@@ -8686,7 +8686,7 @@
   test -z "$as_dir" && as_dir=.
     for ac_exec_ext in '' $ac_executable_extensions; do
   if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_ac_ct_NCURSESW_CONFIG="ncursesw5-config"
+    ac_cv_prog_ac_ct_NCURSESW_CONFIG="ncursesw6-config"
     $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
     break 2
   fi
--- a/configure.ac
+++ b/configure.ac
@@ -488,7 +488,7 @@
 
     if test "x$enable_utf8" != xno; then
 	OLDLIBS="$LIBS"
-	AC_CHECK_TOOL(NCURSESW_CONFIG, ncursesw5-config, no)
+	AC_CHECK_TOOL(NCURSESW_CONFIG, ncursesw6-config, no)
 	if test "x$NCURSESW_CONFIG" != xno; then
 	    CURSES_LIB=`$NCURSESW_CONFIG --libs`
 	    LIBS="$CURSES_LIB $LIBS"
