--- a/src/read.c
+++ b/src/read.c
@@ -111,6 +111,8 @@ static const char *default_include_direc
     INCLUDEDIR,
 #endif
 #ifndef _AMIGA
+    "/opt/include",
+    "/opt/usr/include",
     "/usr/gnu/include",
     "/usr/local/include",
     "/usr/include",
--- a/src/remake.c
+++ b/src/remake.c
@@ -1690,6 +1690,8 @@ library_search (const char *lib, FILE_TI
   static const char *dirs[] =
     {
 #ifndef _AMIGA
+      "/opt/lib",
+      "/opt/usr/lib",
       "/lib",
       "/usr/lib",
 #endif
