--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -8,7 +8,7 @@
 ;Interface ""
 
 # location of the database directory
-;DatabaseDir "/var/lib/vnstat"
+;DatabaseDir "/opt/var/lib/vnstat"
 
 # locale (LC_ALL) ("-" = use system locale)
 ;Locale "-"
@@ -110,10 +110,10 @@
 ;TopDayEntries  20
 
 # how often (in seconds) interface data is updated
-;UpdateInterval 20
+;UpdateInterval 60
 
 # how often (in seconds) interface status changes are checked
-;PollInterval 5
+;PollInterval 30
 
 # how often (in minutes) data is saved to database
 ;SaveInterval 5
@@ -162,10 +162,10 @@
 ;UpdateFileOwner 1
 
 # file used for logging if UseLogging is set to 1
-;LogFile "/var/log/vnstat/vnstat.log"
+;LogFile "/opt/var/log/vnstat/vnstat.log"
 
 # file used as daemon pid / lock file
-;PidFile "/var/run/vnstat/vnstat.pid"
+;PidFile "/opt/var/run/vnstat/vnstat.pid"
 
 # 1 = 64-bit, 0 = 32-bit, -1 = old style logic, -2 = automatic detection
 ;64bitInterfaceCounters -2
