--- a/Makefile
+++ b/Makefile
@@ -4,22 +4,23 @@
 #
 DESTDIR  =
 
-BINPATH  = /usr/bin
-SBINPATH = /usr/sbin
-SCRPATH  = /usr/share/atop
-LOGPATH  = /var/log/atop
-MAN1PATH = /usr/share/man/man1
-MAN5PATH = /usr/share/man/man5
-MAN8PATH = /usr/share/man/man8
-INIPATH  = /etc/init.d
-SYSDPATH = /usr/lib/systemd/system
-CRNPATH  = /etc/cron.d
-ROTPATH  = /etc/logrotate.d
+BINPATH  = /opt/bin
+SBINPATH = /opt/sbin
+SCRPATH  = /opt/share/atop
+LOGPATH  = /opt/var/log/atop
+MAN1PATH = /opt/share/man/man1
+MAN5PATH = /opt/share/man/man5
+MAN8PATH = /opt/share/man/man8
+INIPATH  = /opt/etc/init.d
+SYSDPATH = /opt/lib/systemd/system
+CRNPATH  = /opt/etc/cron.d
+ROTPATH  = /opt/etc/logrotate.d
 PMPATH1  = /usr/lib/pm-utils/sleep.d
 PMPATH2  = /usr/lib64/pm-utils/sleep.d
 PMPATHD  = /usr/lib/systemd/system-sleep
 
-CFLAGS  += -O2 -I. -Wall  	 # -DHTTPSTATS
+CFLAGS   = -O2 -I. -I$(STAGING_DIR)/opt/include -Wall  	 # -DHTTPSTATS
+LDFLAGS  = -L$(STAGING_DIR)/opt/lib -lncursesw -lm -lz -lrt -lpthread
 OBJMOD0  = version.o
 OBJMOD1  = various.o  deviate.o   procdbase.o
 OBJMOD2  = acctproc.o photoproc.o photosyst.o  rawlog.o ifprop.o parseable.o
@@ -32,7 +33,7 @@
 all: 		atop atopsar atopacctd
 
 atop:		atop.o    $(ALLMODS) Makefile
-		$(CC) atop.o $(ALLMODS) -o atop -lncurses -lz -lm -lrt $(LDFLAGS)
+		$(CC) atop.o $(ALLMODS) -o atop $(LDFLAGS)
 
 atopsar:	atop
 		ln -sf atop atopsar
@@ -41,7 +42,7 @@
 		$(CC) atopacctd.o netlink.o -o atopacctd $(LDFLAGS)
 
 netlink.o:	netlink.c
-		$(CC) -I. -Wall -c netlink.c
+		$(CC) -I. $(CFLAGS) -Wall -c netlink.c
 
 clean:
 		rm -f *.o
--- a/acctproc.c
+++ b/acctproc.c
@@ -135,7 +135,7 @@
 #include "acctproc.h"
 #include "atopacctd.h"
 
-#define	ACCTDIR		"/var/cache/atop.d"
+#define	ACCTDIR		"/opt/var/cache/atop.d"
 #define	ACCTFILE	"atop.acct"
 #define	ACCTENV		"ATOPACCT"
 
@@ -159,8 +159,8 @@
 	char		*name;
 	struct stat	stat;
 } pacctadm[] = {
-	{ "/var/log/pacct",		{0, }, },
-	{ "/var/account/pacct",		{0, }, }
+	{ "/opt/var/log/pacct",		{0, }, },
+	{ "/opt/var/account/pacct",	{0, }, }
 };
 
 /*
--- a/atop.c
+++ b/atop.c
@@ -464,7 +464,7 @@
 	/*
 	** read defaults-files /etc/atoprc en $HOME/.atoprc (if any)
 	*/
-	readrc("/etc/atoprc", 1);
+	readrc("/opt/etc/atoprc", 1);
 
 	if ( (p = getenv("HOME")) )
 	{
--- a/atopacctd.c
+++ b/atopacctd.c
@@ -303,7 +303,7 @@
 
 	umask(022);
 
-	(void) chdir("/tmp");			// go to a safe place
+	(void) chdir("/opt/tmp");		// go to a safe place
 
 	/*
 	** increase semaphore to define that atopacctd is running
--- a/atopacctd.h
+++ b/atopacctd.h
@@ -7,7 +7,7 @@
 /*
 ** name of the PID file
 */
-#define	PIDFILE		"/var/run/atopacctd.pid"
+#define	PIDFILE		"/opt/var/run/atopacctd.pid"
 
 /*
 ** directory containing the source accounting file and
@@ -15,7 +15,7 @@
 ** this directory can be overruled by a command line parameter (atopacctd)
 ** or by a keyword in the /etc/atoprc file (atop)
 */
-#define	PACCTDIR	"/var/run"
+#define	PACCTDIR	"/opt/var/run"
 
 /*
 ** accounting file (source file to which kernel writes records)
--- a/netatopd.h
+++ b/netatopd.h
@@ -1,6 +1,6 @@
 #define SEMAKEY         1541961
 
-#define NETEXITFILE     "/var/run/netatop.log"
+#define NETEXITFILE     "/opt/var/run/netatop.log"
 #define MYMAGIC         (unsigned int) 0xfeedb0b0
 
 struct naheader {
--- a/rawlog.c
+++ b/rawlog.c
@@ -50,7 +50,7 @@
 #include "photoproc.h"
 #include "photosyst.h"
 
-#define	BASEPATH	"/var/log/atop"  
+#define	BASEPATH	"/opt/var/log/atop"  
 
 /*
 ** structure which describes the raw file contents
@@ -483,7 +483,7 @@
 		** compressed raw file to be decompressed via gunzip
 		*/
 		fprintf(stderr, "Decompressing logfile ....\n");
-		snprintf(tmpname2, sizeof tmpname2, "/tmp/atopwrkXXXXXX");
+		snprintf(tmpname2, sizeof tmpname2, "/opt/tmp/atopwrkXXXXXX");
 		rawfd = mkstemp(tmpname2);
 		if (rawfd == -1)
 		{
