--- a/acctproc.c
+++ b/acctproc.c
@@ -113,7 +113,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: acctproc.c,v 1.28 2010/04/23 12:20:19 gerlof Exp $";
 
 #define	_FILE_OFFSET_BITS	64
 
--- a/atop.c
+++ b/atop.c
@@ -271,7 +271,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: atop.c,v 1.49 2010/10/23 14:01:00 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/atopsar.c
+++ b/atopsar.c
@@ -29,7 +29,6 @@
 ** --------------------------------------------------------------------------
 */
 
-static const char rcsid[] = "$Id: atopsar.c,v 1.28 2010/11/26 06:19:43 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/deviate.c
+++ b/deviate.c
@@ -168,7 +168,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: deviate.c,v 1.45 2010/10/23 14:02:03 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/photoproc.c
+++ b/photoproc.c
@@ -136,7 +136,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: photoproc.c,v 1.33 2010/04/23 12:19:35 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/photosyst.c
+++ b/photosyst.c
@@ -149,9 +149,8 @@
 **
 */
 
-static const char rcsid[] = "$Id: photosyst.c,v 1.38 2010/11/19 07:40:40 gerlof Exp $";
-
-#include <sys/types.h>
+
+#include <sys/sysmacros.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
--- a/procdbase.c
+++ b/procdbase.c
@@ -58,7 +58,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: procdbase.c,v 1.8 2010/04/23 12:19:35 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/showgeneric.c
+++ b/showgeneric.c
@@ -255,7 +255,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: showgeneric.c,v 1.71 2010/10/25 19:08:32 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/showlinux.c
+++ b/showlinux.c
@@ -262,7 +262,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: showlinux.c,v 1.70 2010/10/23 14:04:12 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
@@ -2011,7 +2010,7 @@
 
         if (acpu < bcpu) return  1;
         if (acpu > bcpu) return -1;
-                         return compmem(a, b);
+        return compmem(a, b);
 }
 
 int
@@ -2035,7 +2034,7 @@
 
         if (adsk < bdsk) return  1;
         if (adsk > bdsk) return -1;
-                         return compcpu(a, b);
+        return compcpu(a, b);
 }
 
 int
@@ -2046,7 +2045,7 @@
 
         if (amem < bmem) return  1;
         if (amem > bmem) return -1;
-                         return  0;
+        return  0;
 }
 
 int
@@ -2063,7 +2062,7 @@
 
         if (anet < bnet) return  1;
         if (anet > bnet) return -1;
-                         return compcpu(a, b);
+        return compcpu(a, b);
 }
 
 int
@@ -2074,7 +2073,7 @@
 
         if (uida > uidb) return  1;
         if (uida < uidb) return -1;
-                         return  0;
+        return  0;
 }
 
 int
@@ -2105,7 +2104,7 @@
 
         if (aidle < bidle) return -1;
         if (aidle > bidle) return  1;
-                           return  0;
+        return  0;
 }
 
 int
@@ -2116,7 +2115,7 @@
 
         if (amsio < bmsio) return  1;
         if (amsio > bmsio) return -1;
-                           return  0;
+        return  0;
 }
 
 int
@@ -2165,14 +2164,14 @@
         {
                 if (afactor < bfactor)  return  1;
                 if (afactor > bfactor)  return -1;
-                                        return  0;
+                return  0;
         }
 
         if (!aspeed && !bspeed)
         {
                 if ((arbyte + asbyte) < (brbyte + bsbyte))      return  1;
                 if ((arbyte + asbyte) > (brbyte + bsbyte))      return -1;
-                                                                return  0;
+                return  0;
         }
 
         if (aspeed)
@@ -2198,7 +2197,7 @@
 
         if (aused < bused) return  1;
         if (aused > bused) return -1;
-                           return  0;
+        return  0;
 }
 
 int
@@ -2212,7 +2211,7 @@
 
         if (aused < bused) return  1;
         if (aused > bused) return -1;
-                           return  0;
+        return  0;
 }
 
 /*
--- a/showprocs.c
+++ b/showprocs.c
@@ -82,7 +82,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: showprocs.c,v 1.15 2011/09/05 11:44:16 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/showsys.c
+++ b/showsys.c
@@ -68,7 +68,6 @@
 **
 */
 
-static const char rcsid[] = "XXXXXX";
 
 #include <sys/types.h>
 #include <sys/param.h>
--- a/various.c
+++ b/various.c
@@ -98,7 +98,6 @@
 **
 */
 
-static const char rcsid[] = "$Id: various.c,v 1.21 2010/11/12 06:16:16 gerlof Exp $";
 
 #include <sys/types.h>
 #include <sys/param.h>
