--- a/libxfs/xfs_fs.h
+++ b/libxfs/xfs_fs.h
@@ -345,7 +345,15 @@ typedef struct xfs_growfs_rt {
  * Structures returned from ioctl XFS_IOC_FSBULKSTAT & XFS_IOC_FSBULKSTAT_SINGLE
  */
 typedef struct xfs_bstime {
+/* __kernel_long_t since Linux 3.4 */
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,4,0)
+	time_t		tv_sec;		/* seconds		*/
+#else
 	__kernel_long_t tv_sec;		/* seconds		*/
+#endif /* __kernel_long_t since Linux 3.4 */
 	__s32		tv_nsec;	/* and nanoseconds	*/
 } xfs_bstime_t;
 
