--- a/configure
+++ b/configure
@@ -11959,7 +11959,7 @@ for ac_func in alarm accept4 setitimer g
  clock confstr close_range copy_file_range ctermid dup3 execv explicit_bzero \
  explicit_memset faccessat fchmod fchmodat fchown fchownat \
  fdwalk fexecve fdopendir fork fpathconf fstatat ftime ftruncate futimesat \
- futimens futimes gai_strerror getentropy \
+ futimens futimes gai_strerror \
  getgrgid_r getgrnam_r \
  getgrouplist getgroups getlogin getloadavg getpeername getpgid getpid \
  getpriority getresuid getresgid getpwent getpwnam_r getpwuid_r getspnam getspent getsid getwd \
--- a/configure.ac
+++ b/configure.ac
@@ -3715,7 +3715,7 @@ AC_CHECK_FUNCS(alarm accept4 setitimer g
  clock confstr close_range copy_file_range ctermid dup3 execv explicit_bzero \
  explicit_memset faccessat fchmod fchmodat fchown fchownat \
  fdwalk fexecve fdopendir fork fpathconf fstatat ftime ftruncate futimesat \
- futimens futimes gai_strerror getentropy \
+ futimens futimes gai_strerror \
  getgrgid_r getgrnam_r \
  getgrouplist getgroups getlogin getloadavg getpeername getpgid getpid \
  getpriority getresuid getresgid getpwent getpwnam_r getpwuid_r getspnam getspent getsid getwd \
