--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -1290,7 +1290,8 @@
        }
 #endif /* AF_NETLINK */
 
-#if defined(AF_VSOCK)
+#include <linux/version.h> 
+#if defined(AF_VSOCK) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
        case AF_VSOCK:
        {
            struct sockaddr_vm *a = (struct sockaddr_vm *) addr;
@@ -1661,7 +1662,7 @@
     }
 #endif /* AF_NETLINK */
 
-#if defined(AF_VSOCK)
+#if defined(AF_VSOCK) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
     case AF_VSOCK:
     {
         struct sockaddr_vm* addr;
@@ -2207,7 +2208,7 @@
     }
 #endif /* AF_NETLINK */
 
-#if defined(AF_VSOCK)
+#if defined(AF_VSOCK) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
        case AF_VSOCK:
        {
            *len_ret = sizeof (struct sockaddr_vm);
--- a/Modules/socketmodule.h
+++ b/Modules/socketmodule.h
@@ -105,6 +105,8 @@
 #include <sys/kern_control.h>
 #endif
 
+#include <linux/version.h> 
+
 #ifdef HAVE_SOCKADDR_ALG
 #include <linux/if_alg.h>
 #ifndef AF_ALG
@@ -114,7 +116,7 @@
 #define SOL_ALG 279
 #endif
 
-#ifdef HAVE_LINUX_VM_SOCKETS_H
+#if defined HAVE_LINUX_VM_SOCKETS_H && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
 # include <linux/vm_sockets.h>
 #else
 # undef AF_VSOCK
@@ -206,7 +208,7 @@
 #ifdef HAVE_SOCKADDR_ALG
     struct sockaddr_alg alg;
 #endif
-#ifdef AF_VSOCK
+#if defined(AF_VSOCK) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
     struct sockaddr_vm vm;
 #endif
 } sock_addr_t;
