--- a/fsck_hfs.tproj/dfalib/SVerify1.c
+++ b/fsck_hfs.tproj/dfalib/SVerify1.c
@@ -61,6 +61,28 @@
 
 static int CompareExtentFileID(const void *first, const void *second);
 
+static size_t strlcpy(char *dst, const char *src, size_t n)
+{
+    if (!n)
+        return strlen(src);
+
+    const char *s = src;
+
+    while (--n > 0)
+        if (!(*dst++ = *s++))
+            break;
+
+    if (!n)
+    {
+        *dst++ = 0;
+        while (*s++)
+            ;
+    }
+
+    return s - src - 1;
+}
+
+
 /*
  * Check if a volume is journaled.  
  *
