--- a/src/uwsc.c
+++ b/src/uwsc.c
@@ -513,7 +513,7 @@
 
             cl->ssl_require_validation = true;
 
-            if (!glob("/etc/ssl/certs/*.crt", 0, NULL, &gl)) {
+            if (!glob("/opt/etc/ssl/certs/*.crt", 0, NULL, &gl)) {
                 for (i = 0; i < gl.gl_pathc; i++)
                     cl->ssl_ops->context_add_ca_crt_file(cl->ssl_ctx, gl.gl_pathv[i]);
                 globfree(&gl);
