--- a/scripts/seaf-gc.sh
+++ b/scripts/seaf-gc.sh
@@ -11,7 +11,7 @@
 seaf_gc_opts=""
 
 export PATH=${INSTALLPATH}/seafile/bin:$PATH
-export SEAFILE_LD_LIBRARY_PATH=${INSTALLPATH}/seafile/lib/:${INSTALLPATH}/seafile/lib64:${LD_LIBRARY_PATH}
+export PYTHONPATH="/opt/share/seafile/seahub/thirdpart:/opt/lib/python2.7"
 
 script_name=$0
 function usage () {
--- a/scripts/setup-seafile-mysql.sh
+++ b/scripts/setup-seafile-mysql.sh
@@ -9,6 +9,8 @@
 SCRIPT=$(readlink -f "$0")
 INSTALLPATH=$(dirname "${SCRIPT}")
 
+export PYTHONPATH="/opt/share/seafile/seafile-server/seahub/thirdpart:/opt/lib/python2.7"
+
 cd "$INSTALLPATH"
 
 python_script=setup-seafile-mysql.py
