--- a/scripts/reset-admin.sh
+++ b/scripts/reset-admin.sh
@@ -35,7 +35,7 @@ function check_python_executable() {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
--- a/scripts/seaf-fsck.sh
+++ b/scripts/seaf-fsck.sh
@@ -23,7 +23,7 @@ function usage () {
 function validate_ccnet_conf_dir () {
     if [[ ! -d ${default_ccnet_conf_dir} ]]; then
         echo "Error: there is no ccnet config directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -32,7 +32,7 @@ function validate_ccnet_conf_dir () {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
--- a/scripts/seaf-fuse.sh
+++ b/scripts/seaf-fuse.sh
@@ -39,7 +39,7 @@ fi
 function validate_ccnet_conf_dir () {
     if [[ ! -d ${default_ccnet_conf_dir} ]]; then
         echo "Error: there is no ccnet config directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -48,7 +48,7 @@ function validate_ccnet_conf_dir () {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
@@ -65,7 +65,7 @@ function validate_already_running () {
 function warning_if_seafile_not_running () {
     if ! pgrep -f "seafile-controller -c ${default_ccnet_conf_dir}" 2>/dev/null 1>&2; then
         echo
-        echo "Warning: seafile-controller not running. Have you run \"./seafile.sh start\" ?"
+        echo "Warning: seafile-controller not running. Have you run \"service seafile-server start\" ?"
         echo
     fi
 }
--- a/scripts/seaf-gc.sh
+++ b/scripts/seaf-gc.sh
@@ -25,7 +25,7 @@ function usage () {
 function validate_ccnet_conf_dir () {
     if [[ ! -d ${default_ccnet_conf_dir} ]]; then
         echo "Error: there is no ccnet config directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -34,7 +34,7 @@ function validate_ccnet_conf_dir () {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
@@ -56,7 +56,7 @@ function check_component_running() {
 
 function validate_already_running () {
     if pid=$(pgrep -f "seafile-controller -c ${default_ccnet_conf_dir}" 2>/dev/null); then
-        echo "seafile server is still running, stop it by \"seafile.sh stop\""
+        echo "seafile server is still running, stop it by \"service seafile-server stop\""
         echo
         exit 1;
     fi
--- a/scripts/seafile.sh
+++ b/scripts/seafile.sh
@@ -51,7 +51,7 @@ function validate_running_user () {
 function validate_ccnet_conf_dir () {
     if [[ ! -d ${default_ccnet_conf_dir} ]]; then
         echo "Error: there is no ccnet config directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -60,7 +60,7 @@ function validate_ccnet_conf_dir () {
 function validate_central_conf_dir () {
     if [[ ! -d ${central_config_dir} ]]; then
         echo "Error: there is no conf/ directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -69,7 +69,7 @@ function validate_central_conf_dir () {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
--- a/scripts/seahub.sh
+++ b/scripts/seahub.sh
@@ -76,7 +76,7 @@ function check_python_executable() {
 function validate_ccnet_conf_dir () {
     if [[ ! -d ${default_ccnet_conf_dir} ]]; then
         echo "Error: there is no ccnet config directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit -1;
     fi
@@ -85,7 +85,7 @@ function validate_ccnet_conf_dir () {
 function validate_seafile_data_dir () {
     if [[ ! -d ${default_seafile_data_dir} ]]; then
         echo "Error: there is no seafile server data directory."
-        echo "Have you run setup-seafile.sh before this?"
+        echo "Have you run setup-seafile before this?"
         echo ""
         exit 1;
     fi
@@ -131,7 +131,7 @@ fi
 function warning_if_seafile_not_running () {
     if ! pgrep -f "seafile-controller -c ${default_ccnet_conf_dir}" 2>/dev/null 1>&2; then
         echo
-        echo "Warning: seafile-controller not running. Have you run \"./seafile.sh start\" ?"
+        echo "Warning: seafile-controller not running. Have you run \"service seafile-server start\" ?"
         echo
         exit 1
     fi
@@ -165,7 +165,7 @@ function start_seahub () {
     sleep 5
     if ! pgrep -f "$gunicorn_exe seahub.wsgi:application -c ${gunicorn_conf}" 2>/dev/null 1>&2; then
         printf "\033[33mError:Seahub failed to start.\033[m\n"
-        echo "Please try to run \"./seahub.sh start\" again"
+        echo "Please try to run \"service seafile-server start\" again"
         exit 1;
     fi
     echo
--- a/scripts/setup-seafile-mysql.py
+++ b/scripts/setup-seafile-mysql.py
@@ -1557,8 +1557,7 @@ def report_success():
 Your seafile server configuration has been finished successfully.
 -----------------------------------------------------------------
 
-run seafile server:     ./seafile.sh { start | stop | restart }
-run seahub  server:     ./seahub.sh  { start <port> | stop | restart <port> }
+run seafile server:     service seafile-server { start | stop | restart }
 
 -----------------------------------------------------------------
 If you are behind a firewall, remember to allow input/output of these tcp ports:
--- a/scripts/setup-seafile.sh
+++ b/scripts/setup-seafile.sh
@@ -716,8 +716,7 @@ echo "----------------------------------
 echo "Your seafile server configuration has been completed successfully." 
 echo "-----------------------------------------------------------------"
 echo 
-echo "run seafile server:     ./seafile.sh { start | stop | restart }"
-echo "run seahub  server:     ./seahub.sh  { start <port> | stop | restart <port> }"
+echo "run seafile server:     service seafile-server { start | stop | restart }"
 echo
 echo "-----------------------------------------------------------------"
 echo "If the server is behind a firewall, remember to open these tcp ports:"
