--- a/common/commit-mgr.h
+++ b/common/commit-mgr.h
@@ -6,6 +6,7 @@
 struct _SeafCommitManager;
 typedef struct _SeafCommit SeafCommit;
 
+#include <sqlite3.h>
 #include <glib/gstdio.h>
 #include "db.h"
 
--- a/server/seafile-session.h
+++ b/server/seafile-session.h
@@ -3,6 +3,7 @@
 #ifndef SEAFILE_SESSION_H
 #define SEAFILE_SESSION_H
 
+#include <sqlite3.h>
 #include <ccnet.h>
 #include <ccnet/cevent.h>
 #include <ccnet/job-mgr.h>
--- a/fuse/seafile-session.h
+++ b/fuse/seafile-session.h
@@ -1,6 +1,7 @@
 #ifndef SEAFILE_SESSION_H
 #define SEAFILE_SESSION_H
 
+#include <sqlite3.h>
 #include <stdint.h>
 #include <glib.h>
