--- a/scripts/setup-seafile.sh
+++ b/scripts/setup-seafile.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-SCRIPT=$(readlink -f "$0")
+SCRIPT=/opt/share/seafile/seafile-server/0.sh
 INSTALLPATH=$(dirname "${SCRIPT}")
 TOPDIR=$(dirname "${INSTALLPATH}")
 default_ccnet_conf_dir=${TOPDIR}/ccnet
@@ -8,7 +8,6 @@
 default_seahub_db=${TOPDIR}/seahub.db
 default_conf_dir=${TOPDIR}/conf
 
-export SEAFILE_LD_LIBRARY_PATH=${INSTALLPATH}/seafile/lib/:${INSTALLPATH}/seafile/lib64:${LD_LIBRARY_PATH}
 
 use_existing_ccnet="false"
 use_existing_seafile="false"
@@ -110,7 +109,7 @@
     if [[ "$PYTHON" != "" && -x $PYTHON ]]; then
         return 0
     fi
-
+        
     if which python2.7 2>/dev/null 1>&2; then
         PYTHON=python2.7
     elif which python27 2>/dev/null 1>&2; then
@@ -339,7 +338,7 @@
     src_docs_dir=${INSTALLPATH}/seafile/docs/
     library_template_dir=${seafile_data_dir}/library-template
     mkdir -p ${library_template_dir}
-    cp -f ${src_docs_dir}/*.doc ${library_template_dir}
+#    cp -f ${src_docs_dir}/*.doc ${library_template_dir}
 }
 
 
@@ -348,12 +347,12 @@
 # -------------------------------------------
 
 check_root;
-sleep .5
+sleep 1
 check_sanity;
 welcome;
-sleep .5
+sleep 1
 check_system_dependency;
-sleep .5
+sleep 1
 
 check_existing_ccnet;
 if [[ ${use_existing_ccnet} != "true" ]]; then
@@ -368,7 +367,7 @@
     get_fileserver_port
 fi
 
-sleep .5
+sleep 1
 
 printf "\nThis is your config information:\n\n"
 
@@ -410,7 +409,7 @@
     echo
 fi
 
-sleep 0.5
+sleep 1
 
 # -------------------------------------------
 # Create seafile conf
@@ -593,7 +592,7 @@
 # copy user manuals to library template
 # -------------------------------------------
 copy_user_manuals;
-
+#
 # -------------------------------------------
 # final message
 # -------------------------------------------
@@ -606,7 +605,9 @@
 echo "-----------------------------------------------------------------"
 echo 
 echo "run seafile server:     ./seafile.sh { start | stop | restart }"
+echo "or        /opt/etc/init.d/S66seafile { start | stop | restart }"
 echo "run seahub  server:     ./seahub.sh  { start <port> | stop | restart <port> }"
+echo "or         /opt/etc/init.d/S67seahub { start | stop | restart }"
 echo
 echo "-----------------------------------------------------------------"
 echo "If the server is behind a firewall, remember to open these tcp ports:"
@@ -615,7 +616,4 @@
 echo "port of seafile fileserver:   ${fileserver_port}"
 echo "port of seahub:               8000"
 echo
-echo -e "When problems occur, refer to\n"
-echo -e "      ${server_manual_http}\n"
-echo "for more information."
 echo
--- a/scripts/seafile.sh
+++ b/scripts/seafile.sh
@@ -12,7 +12,7 @@
 
 echo ""
 
-SCRIPT=$(readlink -f "$0")
+SCRIPT=/opt/share/seafile/seafile-server/0.sh
 INSTALLPATH=$(dirname "${SCRIPT}")
 TOPDIR=$(dirname "${INSTALLPATH}")
 default_ccnet_conf_dir=${TOPDIR}/ccnet
@@ -21,8 +21,6 @@
 
 
 export PATH=${INSTALLPATH}/seafile/bin:$PATH
-export ORIG_LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
-export SEAFILE_LD_LIBRARY_PATH=${INSTALLPATH}/seafile/lib/:${INSTALLPATH}/seafile/lib64:${LD_LIBRARY_PATH}
 
 script_name=$0
 function usage () {
--- a/scripts/seahub.sh
+++ b/scripts/seahub.sh
@@ -12,7 +12,7 @@
 
 echo ""
 
-SCRIPT=$(readlink -f "$0")
+SCRIPT=/opt/share/seafile/seafile-server/0.sh
 INSTALLPATH=$(dirname "${SCRIPT}")
 TOPDIR=$(dirname "${INSTALLPATH}")
 default_ccnet_conf_dir=${TOPDIR}/ccnet
@@ -210,8 +210,7 @@
     export CCNET_CONF_DIR=${default_ccnet_conf_dir}
     export SEAFILE_CONF_DIR=${seafile_data_dir}
     export SEAFILE_CENTRAL_CONF_DIR=${central_config_dir}
-    export PYTHONPATH=${INSTALLPATH}/seafile/lib/python2.6/site-packages:${INSTALLPATH}/seafile/lib64/python2.6/site-packages:${INSTALLPATH}/seahub:${INSTALLPATH}/seahub/thirdpart:$PYTHONPATH
-    export PYTHONPATH=${INSTALLPATH}/seafile/lib/python2.7/site-packages:${INSTALLPATH}/seafile/lib64/python2.7/site-packages:$PYTHONPATH
+    export PYTHONPATH=${INSTALLPATH}/seahub/thirdpart:$PYTHONPATH
 
 }
 
