### Remove the libdrop_ambient admin tool (min. kernel linux-4.3)
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -28,12 +28,11 @@ CLEANFILES = $(BUILT_SOURCES)
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 AM_CFLAGS = -fPIC -DPIC -W -Wall -Wshadow -Wformat -Wundef -D_GNU_SOURCE
 AM_CPPFLAGS = -I. -I${top_srcdir}
-lib_LTLIBRARIES = libcap-ng.la libdrop_ambient.la
+lib_LTLIBRARIES = libcap-ng.la
 include_HEADERS = cap-ng.h
 noinst_HEADERS = captab.h
 libcap_ng_la_SOURCES = cap-ng.c lookup_table.c
 libcap_ng_la_LIBADD =
 libcap_ng_la_DEPENDENCIES = $(libcap_ng_la_SOURCES) ../config.h
 libcap_ng_la_LDFLAGS = -Wl,-z,relro
-libdrop_ambient_la_SOURCES = libdrop_ambient.c
 
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -165,9 +165,6 @@ am__v_lt_1 =
 libcap_ng_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(libcap_ng_la_LDFLAGS) $(LDFLAGS) -o $@
-libdrop_ambient_la_LIBADD =
-am_libdrop_ambient_la_OBJECTS = libdrop_ambient.lo
-libdrop_ambient_la_OBJECTS = $(am_libdrop_ambient_la_OBJECTS)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -184,7 +181,7 @@ DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/cap-ng.Plo \
-	./$(DEPDIR)/libdrop_ambient.Plo ./$(DEPDIR)/lookup_table.Plo
+	./$(DEPDIR)/lookup_table.Plo
 am__mv = mv -f
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
@@ -204,8 +201,8 @@ AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
-SOURCES = $(libcap_ng_la_SOURCES) $(libdrop_ambient_la_SOURCES)
-DIST_SOURCES = $(libcap_ng_la_SOURCES) $(libdrop_ambient_la_SOURCES)
+SOURCES = $(libcap_ng_la_SOURCES)
+DIST_SOURCES = $(libcap_ng_la_SOURCES)
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
 	ctags-recursive dvi-recursive html-recursive info-recursive \
 	install-data-recursive install-dvi-recursive \
@@ -427,14 +424,13 @@ CLEANFILES = $(BUILT_SOURCES)
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 AM_CFLAGS = -fPIC -DPIC -W -Wall -Wshadow -Wformat -Wundef -D_GNU_SOURCE
 AM_CPPFLAGS = -I. -I${top_srcdir}
-lib_LTLIBRARIES = libcap-ng.la libdrop_ambient.la
+lib_LTLIBRARIES = libcap-ng.la
 include_HEADERS = cap-ng.h
 noinst_HEADERS = captab.h
 libcap_ng_la_SOURCES = cap-ng.c lookup_table.c
 libcap_ng_la_LIBADD = 
 libcap_ng_la_DEPENDENCIES = $(libcap_ng_la_SOURCES) ../config.h
 libcap_ng_la_LDFLAGS = -Wl,-z,relro
-libdrop_ambient_la_SOURCES = libdrop_ambient.c
 all: all-recursive
 
 .SUFFIXES:
@@ -509,9 +505,6 @@ clean-libLTLIBRARIES:
 libcap-ng.la: $(libcap_ng_la_OBJECTS) $(libcap_ng_la_DEPENDENCIES) $(EXTRA_libcap_ng_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libcap_ng_la_LINK) -rpath $(libdir) $(libcap_ng_la_OBJECTS) $(libcap_ng_la_LIBADD) $(LIBS)
 
-libdrop_ambient.la: $(libdrop_ambient_la_OBJECTS) $(libdrop_ambient_la_DEPENDENCIES) $(EXTRA_libdrop_ambient_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(LINK) -rpath $(libdir) $(libdrop_ambient_la_OBJECTS) $(libdrop_ambient_la_LIBADD) $(LIBS)
-
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
@@ -519,7 +512,6 @@ distclean-compile:
 	-rm -f *.tab.c
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cap-ng.Plo@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdrop_ambient.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lookup_table.Plo@am__quote@ # am--include-marker
 
 $(am__depfiles_remade):
@@ -801,7 +793,6 @@ clean-am: clean-generic clean-libLTLIBRA
 
 distclean: distclean-recursive
 		-rm -f ./$(DEPDIR)/cap-ng.Plo
-	-rm -f ./$(DEPDIR)/libdrop_ambient.Plo
 	-rm -f ./$(DEPDIR)/lookup_table.Plo
 	-rm -f Makefile
 distclean-am: clean-am distclean-compile distclean-generic \
@@ -849,7 +840,6 @@ installcheck-am:
 
 maintainer-clean: maintainer-clean-recursive
 		-rm -f ./$(DEPDIR)/cap-ng.Plo
-	-rm -f ./$(DEPDIR)/libdrop_ambient.Plo
 	-rm -f ./$(DEPDIR)/lookup_table.Plo
 	-rm -f Makefile
 maintainer-clean-am: distclean-am maintainer-clean-generic
