--- a/CMakeModules/BaseConfig.cmake
+++ b/CMakeModules/BaseConfig.cmake
@@ -10,7 +10,7 @@
 		# this should be addressed at some point in time, but for now, just ignore them.
 		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_BSD_SOURCE -Wno-deprecated-declarations")
 	elseif(UNIX)
-		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_BSD_SOURCE -D_POSIX_C_SOURCE=200112")
+		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112")
 	endif(APPLE)
 
 endif(CMAKE_COMPILER_IS_GNUCC)
