--- a/src/api/api_profile.c
+++ b/src/api/api_profile.c
@@ -21,7 +21,7 @@
 #include "access.h"
 #include "htsmsg.h"
 #include "api.h"
-#include "profile.h"
+#include "../profile.h"
 
 /*
  *
--- a/src/config.c
+++ b/src/config.c
@@ -27,7 +27,7 @@
 #include "htsbuf.h"
 #include "spawn.h"
 #include "lock.h"
-#include "profile.h"
+#include "./profile.h"
 
 /* *************************************************************************
  * Global data
--- a/src/dvr/dvr.h
+++ b/src/dvr/dvr.h
@@ -24,7 +24,7 @@
 #include "channels.h"
 #include "subscriptions.h"
 #include "muxer.h"
-#include "profile.h"
+#include "../profile.h"
 #include "lang_str.h"
 
 typedef struct dvr_config {
--- a/src/input/mpegts/mpegts_mux.c
+++ b/src/input/mpegts/mpegts_mux.c
@@ -24,7 +24,7 @@
 #include "streaming.h"
 #include "channels.h"
 #include "access.h"
-#include "profile.h"
+#include "../../profile.h"
 #include "dvb_charset.h"
 
 #include <assert.h>
--- a/src/input/mpegts/mpegts_mux_sched.c
+++ b/src/input/mpegts/mpegts_mux_sched.c
@@ -22,7 +22,7 @@
 #include "input/mpegts/mpegts_mux_sched.h"
 #include "streaming.h"
 #include "settings.h"
-#include "profile.h"
+#include "../../profile.h"
 
 static void mpegts_mux_sched_timer ( void *p );
 static void mpegts_mux_sched_input ( void *p, streaming_message_t *sm );
--- a/src/main.c
+++ b/src/main.c
@@ -70,7 +70,7 @@
 #if ENABLE_LIBAV
 #include "libav.h"
 #endif
-#include "profile.h"
+#include "./profile.h"
 #include "bouquet.h"
 #include "tvhtime.h"
 
--- a/src/profile.c
+++ b/src/profile.c
@@ -18,7 +18,7 @@
 
 #include "tvheadend.h"
 #include "settings.h"
-#include "profile.h"
+#include "./profile.h"
 #include "streaming.h"
 #include "access.h"
 #include "plumbing/tsfix.h"
--- a/src/satip/rtsp.c
+++ b/src/satip/rtsp.c
@@ -20,7 +20,7 @@
 #include "tvheadend.h"
 #include "htsbuf.h"
 #include "config.h"
-#include "profile.h"
+#include "../profile.h"
 #include "satip/server.h"
 
 #include <ctype.h>
--- a/src/service_mapper.c
+++ b/src/service_mapper.c
@@ -30,7 +30,7 @@
 #include "service_mapper.h"
 #include "streaming.h"
 #include "service.h"
-#include "profile.h"
+#include "./profile.h"
 #include "bouquet.h"
 #include "api.h"
 
--- a/src/subscriptions.c
+++ b/src/subscriptions.c
@@ -36,7 +36,7 @@
 #include "streaming.h"
 #include "channels.h"
 #include "service.h"
-#include "profile.h"
+#include "./profile.h"
 #include "htsmsg.h"
 #include "notify.h"
 #include "atomic.h"
--- a/src/webui/webui.c
+++ b/src/webui/webui.c
@@ -37,7 +37,7 @@
 #include "dvr/dvr.h"
 #include "filebundle.h"
 #include "streaming.h"
-#include "profile.h"
+#include "../profile.h"
 #include "epg.h"
 #include "muxer.h"
 #include "imagecache.h"
