--- a/src/openssl/crypto.c
+++ b/src/openssl/crypto.c
@@ -21,7 +21,12 @@
 
 #include <openssl/rand.h>
 #include <openssl/evp.h>
+
+#ifndef OPENSSL_NO_ENGINE
 #include <openssl/engine.h>
+#else
+#include <openssl/err.h>
+#endif /* OPENSSL_NO_ENGINE */
 
 #include "../crypto.h"
 
@@ -94,8 +99,10 @@ void randomize(void *out, size_t outlen)
 void crypto_init(void) {
 	random_init();
 
+#ifndef OPENSSL_NO_ENGINE
 	ENGINE_load_builtin_engines();
 	ENGINE_register_all_complete();
+#endif /* OPENSSL_NO_ENGINE */
 #if OPENSSL_API_COMPAT < 0x10100000L
 	ERR_load_crypto_strings();
 	OpenSSL_add_all_algorithms();
@@ -111,7 +118,9 @@ void crypto_exit(void) {
 #if OPENSSL_API_COMPAT < 0x10100000L
 	EVP_cleanup();
 	ERR_free_strings();
+#ifndef OPENSSL_NO_ENGINE
 	ENGINE_cleanup();
+#endif /* OPENSSL_NO_ENGINE */
 #endif
 	random_exit();
 }
