--- a/utils/common/media-info.cpp
+++ b/utils/common/media-info.cpp
@@ -21,6 +21,7 @@
 #include <linux/media.h>
 
 #include <iostream>
+#include <fstream>      // std::ifstream
 #include <media-info.h>
 
 static std::string num2s(unsigned num, bool is_hex = true)
--- a/utils/v4l2-compliance/media-info.cpp
+++ b/utils/v4l2-compliance/media-info.cpp
@@ -21,6 +21,7 @@
 #include <linux/media.h>
 
 #include <iostream>
+#include <fstream>      // std::ifstream
 #include <media-info.h>
 
 static std::string num2s(unsigned num, bool is_hex = true)
--- a/utils/v4l2-ctl/media-info.cpp
+++ b/utils/v4l2-ctl/media-info.cpp
@@ -21,6 +21,7 @@
 #include <linux/media.h>
 
 #include <iostream>
+#include <fstream>      // std::ifstream
 #include <media-info.h>
 
 static std::string num2s(unsigned num, bool is_hex = true)
