--- a/src/util/sys_defs.h
+++ b/src/util/sys_defs.h
@@ -766,18 +766,18 @@ extern int initgroups(const char *, int)
 #define HAS_NIS
 #endif
 #define GETTIMEOFDAY(t)	gettimeofday(t,(struct timezone *) 0)
-#define ROOT_PATH	"/bin:/usr/bin:/sbin:/usr/sbin"
+#define ROOT_PATH	"/opt/bin:/opt/sbin:/bin:/usr/bin:/sbin:/usr/sbin"
 #define FIONREAD_IN_TERMIOS_H
 #define USE_STATFS
 #define STATFS_IN_SYS_VFS_H
 #define PREPEND_PLUS_TO_OPTSTRING
 #define HAS_POSIX_REGEXP
 #define HAS_DLOPEN
-#define NATIVE_SENDMAIL_PATH "/usr/sbin/sendmail"
-#define NATIVE_MAILQ_PATH "/usr/bin/mailq"
-#define NATIVE_NEWALIAS_PATH "/usr/bin/newaliases"
-#define NATIVE_COMMAND_DIR "/usr/sbin"
-#define NATIVE_DAEMON_DIR "/usr/libexec/postfix"
+#define NATIVE_SENDMAIL_PATH "/opt/sbin/sendmail"
+#define NATIVE_MAILQ_PATH "/opt/bin/mailq"
+#define NATIVE_NEWALIAS_PATH "/opt/bin/newaliases"
+#define NATIVE_COMMAND_DIR "/opt/sbin"
+#define NATIVE_DAEMON_DIR "/opt/libexec/postfix"
 #ifdef __GLIBC_PREREQ
 #define HAVE_GLIBC_API_VERSION_SUPPORT(maj, min) __GLIBC_PREREQ(maj, min)
 #else
--- a/src/global/mail_params.h
+++ b/src/global/mail_params.h
@@ -279,7 +279,7 @@ extern int var_smtp_mxsess_limit;
   */
 #define VAR_QUEUE_DIR	"queue_directory"
 #ifndef DEF_QUEUE_DIR
-#define DEF_QUEUE_DIR	"/var/spool/postfix"
+#define DEF_QUEUE_DIR	"/opt/var/spool/postfix"
 #endif
 extern char *var_queue_dir;
 
@@ -288,13 +288,13 @@ extern char *var_queue_dir;
   */
 #define VAR_DAEMON_DIR		"daemon_directory"
 #ifndef DEF_DAEMON_DIR
-#define DEF_DAEMON_DIR		"/usr/libexec/postfix"
+#define DEF_DAEMON_DIR		"/opt/libexec/postfix"
 #endif
 extern char *var_daemon_dir;
 
 #define VAR_COMMAND_DIR		"command_directory"
 #ifndef DEF_COMMAND_DIR
-#define DEF_COMMAND_DIR		"/usr/sbin"
+#define DEF_COMMAND_DIR		"/opt/sbin"
 #endif
 extern char *var_command_dir;
 
@@ -312,7 +312,7 @@ extern char *var_pid_dir;
   */
 #define VAR_DATA_DIR		"data_directory"
 #ifndef DEF_DATA_DIR
-#define DEF_DATA_DIR		"/var/lib/postfix"
+#define DEF_DATA_DIR		"/opt/var/lib/postfix"
 #endif
 extern char *var_data_dir;
 
@@ -326,7 +326,7 @@ extern time_t var_starttime;
   */
 #define VAR_CONFIG_DIR		"config_directory"
 #ifndef DEF_CONFIG_DIR
-#define DEF_CONFIG_DIR		"/etc/postfix"
+#define DEF_CONFIG_DIR		"/opt/etc/postfix"
 #endif
 extern char *var_config_dir;
 
@@ -571,7 +571,7 @@ extern char *var_luser_relay;
   */
 #define VAR_MAIL_SPOOL_DIR	"mail_spool_directory"
 #ifndef DEF_MAIL_SPOOL_DIR
-#define DEF_MAIL_SPOOL_DIR	_PATH_MAILDIR
+#define DEF_MAIL_SPOOL_DIR	"/opt/var/mail"
 #endif
 extern char *var_mail_spool_dir;
 
@@ -2689,17 +2689,17 @@ extern int var_fault_inj_code;
   */
 #define VAR_SENDMAIL_PATH		"sendmail_path"
 #ifndef DEF_SENDMAIL_PATH
-#define DEF_SENDMAIL_PATH		"/usr/sbin/sendmail"
+#define DEF_SENDMAIL_PATH		"/opt/sbin/sendmail"
 #endif
 
 #define VAR_MAILQ_PATH			"mailq_path"
 #ifndef DEF_MAILQ_PATH
-#define DEF_MAILQ_PATH			"/usr/bin/mailq"
+#define DEF_MAILQ_PATH			"/opt/bin/mailq"
 #endif
 
 #define VAR_NEWALIAS_PATH		"newaliases_path"
 #ifndef DEF_NEWALIAS_PATH
-#define DEF_NEWALIAS_PATH		"/usr/bin/newaliases"
+#define DEF_NEWALIAS_PATH		"/opt/bin/newaliases"
 #endif
 
 #define VAR_OPENSSL_PATH		"openssl_path"
@@ -2710,7 +2710,7 @@ extern char *var_openssl_path;
 
 #define VAR_MANPAGE_DIR			"manpage_directory"
 #ifndef DEF_MANPAGE_DIR
-#define DEF_MANPAGE_DIR			"/usr/local/man"
+#define DEF_MANPAGE_DIR			"/opt/local/man"
 #endif
 
 #define VAR_SAMPLE_DIR			"sample_directory"
@@ -4110,7 +4110,7 @@ extern bool var_smtp_tls_conn_reuse;
   */
 #define VAR_SHLIB_DIR	"shlib_directory"
 #ifndef DEF_SHLIB_DIR
-#define DEF_SHLIB_DIR	"/usr/lib/postfix"
+#define DEF_SHLIB_DIR	"/opt/lib/postfix"
 #endif
 extern char *var_shlib_dir;
 
