--- a/bind/Makefile.in
+++ b/bind/Makefile.in
@@ -96,7 +96,7 @@
 		echo Building BIND Export libraries - this takes some time. ;\
 		(cd ${bindsrcdir}/lib/export/dns ;			     \
 		  echo building gen using ${BUILD_CC} in `pwd` ;             \
-		  $(MAKE) CC=${BUILD_CC} CFLAGS="-O2" LIBS="" gen) ;         \
+		  $(MAKE) CC=${BUILD_CC} CFLAGS="-O2" LIBS="" LDFLAGS="" gen) ;\
                                                                              \
 		echo Installing BIND Export libraries to ${binddir}. ;       \
 		(cd ${bindsrcdir}/lib/export ;                               \
