--- a/glib/gthread-posix.c
+++ b/glib/gthread-posix.c
@@ -74,6 +74,11 @@
 #include <sys/syscall.h>
 #endif
 
+/* __kernel_long_t since Linux 3.4 */
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
 #if (defined(HAVE_FUTEX) || defined(HAVE_FUTEX_TIME64)) && \
     (defined(HAVE_STDATOMIC_H) || defined(__ATOMIC_SEQ_CST))
 #define USE_NATIVE_MUTEX
@@ -1649,7 +1654,7 @@ g_cond_wait_until (GCond  *cond,
   }
 #endif
 
-#ifdef __NR_futex
+#if defined(__NR_futex) && LINUX_VERSION_CODE >= KERNEL_VERSION(3,4,0)
   {
 #  ifdef __kernel_long_t
 #    define KERNEL_SPAN_SEC_TYPE __kernel_long_t
