--- a/glib/gtimezone.c
+++ b/glib/gtimezone.c
@@ -515,13 +515,13 @@ zone_identifier_illumos (void)
 static const gchar *
 zone_info_base_dir (void)
 {
-  if (g_file_test ("/usr/share/zoneinfo", G_FILE_TEST_IS_DIR))
-    return "/usr/share/zoneinfo";     /* Most distros */
+  if (g_file_test ("/opt/share/zoneinfo", G_FILE_TEST_IS_DIR))
+    return "/opt/share/zoneinfo";     /* Most distros */
   else if (g_file_test ("/usr/share/lib/zoneinfo", G_FILE_TEST_IS_DIR))
     return "/usr/share/lib/zoneinfo"; /* Illumos distros */
 
   /* need a better fallback case */
-  return "/usr/share/zoneinfo";
+  return "/opt/share/zoneinfo";
 }
 
 static gchar *
@@ -534,7 +534,7 @@ zone_identifier_unix (void)
   const gchar *tzdir;
 
   /* Resolve the actual timezone pointed to by /etc/localtime. */
-  resolved_identifier = g_file_read_link ("/etc/localtime", &read_link_err);
+  resolved_identifier = g_file_read_link ("/opt/etc/localtime", &read_link_err);
   if (resolved_identifier == NULL)
     {
       gboolean not_a_symlink = g_error_matches (read_link_err,
@@ -573,7 +573,7 @@ zone_identifier_unix (void)
   else
     {
       /* Resolve relative path */
-      canonical_path = g_canonicalize_filename (resolved_identifier, "/etc");
+      canonical_path = g_canonicalize_filename (resolved_identifier, "/opt/etc");
       g_free (resolved_identifier);
       resolved_identifier = g_steal_pointer (&canonical_path);
     }
@@ -634,7 +634,7 @@ zone_info_unix (const gchar *identifier,
       if (resolved_identifier == NULL)
         goto out;
 
-      filename = g_strdup ("/etc/localtime");
+      filename = g_strdup ("/opt/etc/localtime");
     }
 
   file = g_mapped_file_new (filename, FALSE, NULL);
