--- a/gio/xdgmime/xdgmime.c
+++ b/gio/xdgmime/xdgmime.c
@@ -228,7 +228,7 @@ xdg_init_dirs (void)
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
 
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "/opt/usr/share/:/opt/share/:/usr/local/share/:/usr/share/";
 
   /* Work out how many dirs we’re dealing with. */
   if (xdg_data_home != NULL || home != NULL)
--- a/glib/gcharset.c
+++ b/glib/gcharset.c
@@ -503,7 +503,7 @@ unalias_lang (char *lang)
   if (g_once_init_enter_pointer (&alias_table))
     {
       GHashTable *table = g_hash_table_new (g_str_hash, g_str_equal);
-      read_aliases ("/usr/share/locale/locale.alias", table);
+      read_aliases ("/opt/share/locale/locale.alias", table);
       g_once_init_leave_pointer (&alias_table, table);
     }
 
--- a/glib/gspawn-posix.c
+++ b/glib/gspawn-posix.c
@@ -1404,7 +1404,7 @@ fork_exec (gboolean              interme
        * * what to search if PATH is unset. POSIX may, dunno.
        * */
 
-      chosen_search_path = "/bin:/usr/bin:.";
+      chosen_search_path = "/opt/bin:/opt/usr/bin:/bin:/usr/bin:.";
     }
 
   if (search_path || search_path_from_envp)
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -385,7 +385,7 @@ g_find_program_for_path (const char *pro
        * what to search if PATH is unset. POSIX may, dunno.
        */
       
-      path = "/bin:/usr/bin:.";
+      path = "/opt/bin:/opt/usr/bin:/bin:/usr/bin:.";
     }
 #else
   if G_LIKELY (original_path == NULL)
@@ -1680,7 +1680,7 @@ g_get_os_info (const gchar *key_name)
   else
     return NULL;
 #elif defined (G_OS_UNIX)
-  const gchar * const os_release_files[] = { "/etc/os-release", "/usr/lib/os-release" };
+  const gchar * const os_release_files[] = { "/opt/etc/os-release", "/etc/os-release", "/usr/lib/os-release" };
   gsize i;
   gchar *buffer = NULL;
   gchar *result = NULL;
@@ -2695,7 +2695,7 @@ g_build_system_data_dirs (void)
    */
 #ifndef G_OS_WIN32
   if (!data_dirs || !data_dirs[0])
-    data_dirs = "/usr/local/share/:/usr/share/";
+    data_dirs = "/opt/usr/share/:/opt/share/:/usr/local/share/:/usr/share/";
 
   data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #else
