--- a/src/backtrace_posix.cc
+++ a/src/backtrace_posix.cc
@@ -4,13 +4,7 @@
 #include <features.h>
 #endif
 
-#if defined(__linux__) && !defined(__GLIBC__) || \
-    defined(__UCLIBC__) || \
-    defined(_AIX)
-#define HAVE_EXECINFO_H 0
-#else
-#define HAVE_EXECINFO_H 1
-#endif
+#undef HAVE_EXECINFO_H
 
 #if HAVE_EXECINFO_H
 #include <cxxabi.h>
--- a/src/inspector_socket.cc
+++ b/src/inspector_socket.cc
@@ -530,6 +530,8 @@
   }
 }
 
+extern "C" void http_parser_settings_init(http_parser_settings *settings);
+
 static void init_handshake(inspector_socket_t* inspector) {
   http_parsing_state_s* state = inspector->http_parsing_state;
   CHECK_NE(state, nullptr);
