--- a/utils/statd/statd.c
+++ b/utils/statd/statd.c
@@ -160,7 +160,7 @@ usage(void)
 	fprintf(stderr,"      -H                   Specify a high-availability callout program.\n");
 }
 
-static const char *pidfile = "/var/run/rpc.statd.pid";
+static const char *pidfile = "/opt/var/run/rpc.statd.pid";
 
 int pidfd = -1;
 static void create_pidfile(void)
@@ -195,7 +195,7 @@ static void run_sm_notify(int outport)
 	char *av[6];
 	int ac = 0;
 
-	av[ac++] = "/usr/sbin/sm-notify";
+	av[ac++] = "/opt/sbin/sm-notify";
 	if (run_mode & MODE_NODAEMON)
 		av[ac++] = "-d";
 	if (outport) {
@@ -399,7 +399,7 @@ int main (int argc, char **argv)
 	}
 
 	if (run_mode & MODE_NOTIFY_ONLY) {
-		fprintf(stderr, "%s: -N deprecated, consider using /usr/sbin/sm-notify directly\n",
+		fprintf(stderr, "%s: -N deprecated, consider using /opt/sbin/sm-notify directly\n",
 			name_p);
 		run_sm_notify(out_port);
 	}
--- a/utils/mount/Makefile.am
+++ b/utils/mount/Makefile.am
@@ -2,7 +2,7 @@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = @sbindir@
 
 man8_MANS	= mount.nfs.man umount.nfs.man
 man5_MANS	= nfs.man
--- a/utils/mount/Makefile.in
+++ b/utils/mount/Makefile.in
@@ -425,7 +425,7 @@ psdir = @psdir@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@
