--- a/utils/blkmapd/device-discovery.c
+++ b/utils/blkmapd/device-discovery.c
@@ -64,7 +64,7 @@
 #define EVENT_BUFSIZE (1024 * EVENT_SIZE)
 
 #define RPCPIPE_DIR	"/var/lib/nfs/rpc_pipefs"
-#define PID_FILE	"/tmp/run/blkmapd.pid"
+#define PID_FILE	"/opt/run/blkmapd.pid"
 
 #define CONF_SAVE(w, f) do {			\
 	char *p = f;				\
--- a/utils/gssd/gssd.h
+++ b/utils/gssd/gssd.h
@@ -44,7 +44,7 @@
 #define DNOTIFY_SIGNAL		(SIGRTMIN + 3)
 
 #define GSSD_DEFAULT_CRED_DIR			"/tmp"
-#define GSSD_USER_CRED_DIR			"/tmp/run/user/%U"
+#define GSSD_USER_CRED_DIR			"/opt/run/user/%U"
 #define GSSD_DEFAULT_CRED_PREFIX		"krb5cc"
 #define GSSD_DEFAULT_MACHINE_CRED_SUFFIX	"machine"
 #define GSSD_DEFAULT_KEYTAB_FILE		"/etc/krb5.keytab"
--- a/utils/statd/sm-notify.c
+++ b/utils/statd/sm-notify.c
@@ -913,7 +913,7 @@ static int record_pid(void)
 	int fd;
 
 	(void)snprintf(pid, sizeof(pid), "%d\n", (int)getpid());
-	fd = open("/tmp/run/sm-notify.pid", O_CREAT|O_EXCL|O_WRONLY, 0600);
+	fd = open("/opt/run/sm-notify.pid", O_CREAT|O_EXCL|O_WRONLY, 0600);
 	if (fd < 0)
 		return 0;
 
--- a/utils/statd/statd.c
+++ b/utils/statd/statd.c
@@ -161,7 +161,7 @@ usage(void)
 	fprintf(stderr,"      -H                   Specify a high-availability callout program.\n");
 }
 
-static const char *pidfile = "/tmp/run/rpc.statd.pid";
+static const char *pidfile = "/opt/var/run/rpc.statd.pid";
 
 int pidfd = -1;
 static void create_pidfile(void)
@@ -196,7 +196,7 @@ static void run_sm_notify(int outport)
 	char *av[6];
 	int ac = 0;
 
-	av[ac++] = "/usr/sbin/sm-notify";
+	av[ac++] = "/opt/sbin/sm-notify";
 	if (run_mode & MODE_NODAEMON)
 		av[ac++] = "-d";
 	if (outport) {
@@ -400,7 +400,7 @@ int main (int argc, char **argv)
 	}
 
 	if (run_mode & MODE_NOTIFY_ONLY) {
-		fprintf(stderr, "%s: -N deprecated, consider using /usr/sbin/sm-notify directly\n",
+		fprintf(stderr, "%s: -N deprecated, consider using /opt/sbin/sm-notify directly\n",
 			name_p);
 		run_sm_notify(out_port);
 	}
--- a/utils/mount/Makefile.am
+++ b/utils/mount/Makefile.am
@@ -2,7 +2,7 @@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = @sbindir@
 
 man8_MANS	= mount.nfs.man umount.nfs.man
 man5_MANS	= nfs.man
--- a/utils/mount/Makefile.in
+++ b/utils/mount/Makefile.in
@@ -426,7 +426,7 @@ psdir = @psdir@
 
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time.
-sbindir = /sbin
+sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@
