--- a/CMakeLists.txt
+++ b/CMakeLists.txt.new
@@ -60,7 +60,6 @@
 tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" OFF)
 tr_auto_option(USE_SYSTEM_EVENT2 "Use system event2 library" AUTO)
 tr_auto_option(USE_SYSTEM_DEFLATE "Use system deflate library" AUTO)
-tr_auto_option(USE_SYSTEM_DHT "Use system dht library" AUTO)
 tr_auto_option(USE_SYSTEM_MINIUPNPC "Use system miniupnpc library" AUTO)
 tr_auto_option(USE_SYSTEM_NATPMP "Use system natpmp library" AUTO)
 tr_auto_option(USE_SYSTEM_UTP "Use system utp library" AUTO)
@@ -504,9 +503,7 @@
 
 add_subdirectory(third-party/jsonsl)
 add_subdirectory(third-party/wildmat)
-
-tr_add_external_auto_library(DHT dht dht
-    TARGET dht::dht)
+add_subdirectory(third-party/dht)
 
 tr_add_external_auto_library(PSL libpsl psl
     TARGET psl::psl)
--- a/libtransmission/CMakeLists.txt
+++ b/libtransmission/CMakeLists.txt
@@ -280,7 +280,6 @@
         psl::psl
         natpmp::natpmp
         miniupnpc::libminiupnpc
-        dht::dht
         libutp::libutp
         libb64::libb64
         ${LIBINTL_LIBRARY}
@@ -290,6 +289,7 @@
         jsonsl
         utf8::cpp
         wildmat
+	dht
         WideInteger::WideInteger
         $<$<BOOL:${WIN32}>:crypt32>
         $<$<BOOL:${WIN32}>:shlwapi>
