--- a/conf/zabbix_agentd.conf
+++ b/conf/zabbix_agentd.conf
@@ -3,11 +3,12 @@
 
 ############ GENERAL PARAMETERS #################
 
-# Zabbix always creates a PidFile. Make sure it is where we want it.
-PidFile=/var/run/zabbix-agent/zabbix_agentd.pid
-
-# use syslog
-LogType=system
+### Option: PidFile
+#	Name of PID file.
+#
+# Mandatory: no
+# Default:
+PidFile=/opt/var/run/zabbix-agent/zabbix_agentd.pid
 
 ### Option: LogType
 #	Specifies where log messages are written to:
@@ -24,7 +25,8 @@ LogType=system
 #
 # Mandatory: yes, if LogType is set to file, otherwise no
 # Default:
-# LogFile=
+
+LogFile=/opt/var/log/zabbix-agent/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -540,4 +542,4 @@ User=zabbix-agent
 
 # Include=/usr/local/etc/zabbix_agentd.userparams.conf
 # Include=/usr/local/etc/zabbix_agentd.conf.d/*.conf
-Include=/etc/zabbix_agentd.conf.d/
+Include=/opt/etc/zabbix_agentd.conf.d/
--- a/conf/zabbix_server.conf
+++ b/conf/zabbix_server.conf
@@ -27,7 +27,6 @@
 # Mandatory: no
 # Default:
 # LogType=file
-LogType=system
 
 ### Option: LogFile
 #	Log file name for LogType 'file' parameter.
@@ -36,7 +35,7 @@ LogType=system
 # Default:
 # LogFile=
 
-# LogFile=/tmp/zabbix_server.log
+LogFile=/opt/var/log/zabbix-server/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -46,7 +45,6 @@ LogType=system
 # Range: 0-1024
 # Default:
 # LogFileSize=1
-LogFileSize=0
 
 ### Option: DebugLevel
 #	Specifies debug level:
@@ -71,7 +69,7 @@ LogFileSize=0
 
 # Although procd does not require a pid file, zabbix uses the pidfile to
 # shut down correctly on receipt of a TERM or INT signal.
-PidFile=/var/run/zabbix-server/zabbix_server.pid
+PidFile=/opt/var/run/zabbix-server/zabbix_server.pid
 
 ### Option: SocketDir
 #	IPC socket directory.
--- a/ui/locale/add_new_language.sh
+++ b/ui/locale/add_new_language.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/opt/bin/bash
 
 function run
 {
--- a/ui/locale/make_mo.sh
+++ b/ui/locale/make_mo.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/opt/bin/bash
 
 while read pofile; do
         msgfmt --use-fuzzy -c -o ${pofile%po}mo $pofile || exit $?
--- a/ui/locale/update_po.sh
+++ b/ui/locale/update_po.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/opt/bin/bash
 function show_help
 {
 	echo "$(basename $0) [--cleanup]"
--- a/ui/vendor/bin/yaml-lint
+++ b/ui/vendor/bin/yaml-lint
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/bin/sh
 
 dir=$(cd "${0%[/\\]*}" > /dev/null; cd "../symfony/yaml/Resources/bin" && pwd)
 
--- a/ui/vendor/symfony/yaml/Resources/bin/yaml-lint
+++ b/ui/vendor/symfony/yaml/Resources/bin/yaml-lint
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/opt/bin/php
 <?php
 
 /*
