--- a/tayga.c
+++ b/tayga.c
@@ -118,12 +118,19 @@ static void tun_setup(int do_mktun, int
 					strerror(errno));
 			exit(1);
 		}
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* TUNSETGROUP since Linux 2.6.23 */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,23)
 		if (ioctl(gcfg->tun_fd, TUNSETGROUP, 0) < 0) {
 			slog(LOG_CRIT, "Unable to set group on %s, "
 					"aborting: %s\n", gcfg->tundev,
 					strerror(errno));
 			exit(1);
 		}
+#endif
 		slog(LOG_NOTICE, "Created persistent tun device %s\n",
 				gcfg->tundev);
 		return;
