--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -72,11 +72,11 @@
 
 ccnet-object.h: ${ccnet_object_define}
 	rm -f $@
-	valac --pkg posix ${ccnet_object_define} -C -H ccnet-object.h
+	LC_ALL= LANG=en_US.UTF-8 valac --pkg posix ${ccnet_object_define} -C -H ccnet-object.h
 
 ccnetobj.c: ${ccnet_object_define}
 	rm -f $@
-	valac -C --pkg posix ${ccnet_object_define}
+	LC_ALL= LANG=en_US.UTF-8 valac -C --pkg posix ${ccnet_object_define}
 
 searpc_gen = searpc-signature.h searpc-marshal.h
 
