--- a/lib/shell.c
+++ b/lib/shell.c
@@ -185,6 +185,7 @@
          * so we need to check busybox symlinks *before* checking for the name "bash"
          * in order to avoid that case. */
         mc_shell->type = SHELL_ASH_BUSYBOX;
+        mc_global.tty.use_subshell = FALSE;
         mc_shell->name = mc_shell->path;
     }
     else
@@ -210,6 +211,7 @@
     else if (strstr (mc_shell->path, "/ash") != NULL || getenv ("ASH") != NULL)
     {
         mc_shell->type = SHELL_ASH_BUSYBOX;
+        mc_global.tty.use_subshell = FALSE;
         mc_shell->name = "ash";
     }
     else
