--- a/configure.ac
+++ b/configure.ac
@@ -76,17 +76,17 @@ AM_CONDITIONAL(ICONV, test "$am_cv_func_
 AC_SUBST(libfuse_libs)
 
 if test -z "$MOUNT_FUSE_PATH"; then
-	MOUNT_FUSE_PATH=/sbin
+	MOUNT_FUSE_PATH=/opt/sbin
 	AC_MSG_NOTICE([MOUNT_FUSE_PATH env var not set, using default $MOUNT_FUSE_PATH])
 fi
 AC_SUBST(MOUNT_FUSE_PATH)
 if test -z "$UDEV_RULES_PATH"; then
-	UDEV_RULES_PATH=/etc/udev/rules.d
+	UDEV_RULES_PATH=/opt/etc/udev/rules.d
 	AC_MSG_NOTICE([UDEV_RULES_PATH env var not set, using default $UDEV_RULES_PATH])
 fi
 AC_SUBST(UDEV_RULES_PATH)
 if test -z "$INIT_D_PATH"; then
-	INIT_D_PATH=/etc/init.d
+	INIT_D_PATH=/opt/etc/init.d
 	AC_MSG_NOTICE([INIT_D_PATH env var not set, using default $INIT_D_PATH])
 fi
 AC_SUBST(INIT_D_PATH)
