--- a/util/Makefile.am
+++ b/util/Makefile.am
@@ -22,11 +22,11 @@ ulockmgr_server_LDFLAGS = -pthread
 
 install-exec-hook:
 	-chmod u+s $(DESTDIR)$(bindir)/fusermount
-	@if test ! -e $(DESTDIR)/dev/fuse; then \
-		$(MKDIR_P) $(DESTDIR)/dev; \
-		echo "mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true"; \
-		mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true; \
-	fi
+#	@if test ! -e $(DESTDIR)/dev/fuse; then \
+#		$(MKDIR_P) $(DESTDIR)/dev; \
+#		echo "mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true"; \
+#		mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true; \
+#	fi
 
 EXTRA_DIST = udev.rules init_script
 
@@ -39,10 +39,10 @@ install-exec-local:
 	$(INSTALL_PROGRAM) $(builddir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	$(MKDIR_P) $(DESTDIR)$(INIT_D_PATH)
 	$(INSTALL_SCRIPT) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
-		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
-	fi
+#	@if test -x /usr/sbin/update-rc.d; then \
+#		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
+#		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
+#	fi
 
 install-data-local:
 	$(MKDIR_P) $(DESTDIR)$(UDEV_RULES_PATH)
@@ -52,7 +52,7 @@ uninstall-local:
 	rm -f $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	rm -f $(DESTDIR)$(UDEV_RULES_PATH)/99-fuse.rules
 	rm -f $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse remove || true"; \
-		/usr/sbin/update-rc.d fuse remove || true; \
-	fi
+#	@if test -x /usr/sbin/update-rc.d; then \
+#		echo "/usr/sbin/update-rc.d fuse remove || true"; \
+#		/usr/sbin/update-rc.d fuse remove || true; \
+#	fi
--- a/util/Makefile.in
+++ b/util/Makefile.in
@@ -736,21 +736,21 @@ mount_util.c: $(top_srcdir)/lib/mount_ut
 
 install-exec-hook:
 	-chmod u+s $(DESTDIR)$(bindir)/fusermount
-	-@if test ! -e $(DESTDIR)/dev/fuse; then \
-		$(MKDIR_P) $(DESTDIR)/dev; \
-		echo "mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true"; \
-		mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true; \
-	fi
+#	-@if test ! -e $(DESTDIR)/dev/fuse; then \
+#		$(MKDIR_P) $(DESTDIR)/dev; \
+#		echo "mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true"; \
+#		mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229 || true; \
+#	fi
 
 install-exec-local:
 	$(MKDIR_P) $(DESTDIR)$(MOUNT_FUSE_PATH)
 	$(INSTALL_PROGRAM) $(builddir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	$(MKDIR_P) $(DESTDIR)$(INIT_D_PATH)
 	$(INSTALL_SCRIPT) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
-		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
-	fi
+#	@if test -x /usr/sbin/update-rc.d; then \
+#		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
+#		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
+#	fi
 
 install-data-local:
 	$(MKDIR_P) $(DESTDIR)$(UDEV_RULES_PATH)
@@ -760,10 +760,10 @@ uninstall-local:
 	rm -f $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	rm -f $(DESTDIR)$(UDEV_RULES_PATH)/99-fuse.rules
 	rm -f $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse remove || true"; \
-		/usr/sbin/update-rc.d fuse remove || true; \
-	fi
+#	@if test -x /usr/sbin/update-rc.d; then \
+#		echo "/usr/sbin/update-rc.d fuse remove || true"; \
+#		/usr/sbin/update-rc.d fuse remove || true; \
+#	fi
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
