--- a/plugins/input_uvc/v4l2uvc.c	2018-10-25 09:34:44.000000000 +0300
+++ b/plugins/input_uvc/v4l2uvc.c	2019-04-16 16:01:31.782046436 +0300
@@ -29,6 +29,13 @@
 #include "huffman.h"
 #include "dynctrl.h"
 
+#ifndef V4L2_EVENT_SOURCE_CHANGE
+#define V4L2_EVENT_SOURCE_CHANGE 5
+#endif
+#ifndef VIDIOC_QUERY_DV_TIMINGS
+#define VIDIOC_QUERY_DV_TIMINGS  _IOR('V', 99, struct v4l2_dv_timings)
+#endif
+
 static int debug = 0;
 
 /* fcc2s - convert pixelformat to string
