--- a/plugins/input_uvc/v4l2uvc.c
+++ b/plugins/input_uvc/v4l2uvc.c
@@ -29,6 +29,10 @@
 #include "huffman.h"
 #include "dynctrl.h"
 
+#ifndef V4L2_EVENT_SOURCE_CHANGE
+#define V4L2_EVENT_SOURCE_CHANGE 5
+#endif
+
 static int debug = 0;
 
 /* fcc2s - convert pixelformat to string
@@ -524,7 +528,7 @@ int video_set_dv_timings(struct vdIn *vd
     v4l2_std_id std;
 
     memset(&timings, 0, sizeof(timings));
-    if (xioctl(vd->fd, VIDIOC_QUERY_DV_TIMINGS, &timings) >= 0) {
+    if (xioctl(vd->fd, VIDIOC_S_DV_TIMINGS, &timings) >= 0) {
         IPRINT("QUERY_DV_TIMINGS returned %ux%u pixclk %llu\n", timings.bt.width, timings.bt.height, timings.bt.pixelclock);
         // Can read DV timings, so set them.
         if (xioctl(vd->fd, VIDIOC_S_DV_TIMINGS, &timings) < 0) {
--- a/mjpg_streamer.c
+++ b/mjpg_streamer.c
@@ -255,7 +255,7 @@ int main(int argc, char *argv[])
 
     /* fork to the background */
     if(daemon) {
-        LOG("enabling daemon mode");
+        syslog(LOG_INFO, "enabling daemon mode");
         daemon_mode();
     }
 
--- a/utils.c
+++ b/utils.c
@@ -63,7 +63,7 @@ void daemon_mode(void)
         exit(1);
     }
     if(fr > 0) {
-        fprintf(stderr, "forked to background (%d)\n", fr);
+/*        fprintf(stderr, "forked to background (%d)\n", fr); */
         exit(0);
     }
 
