--- a/src/loop.c
+++ b/src/loop.c
@@ -37,7 +37,7 @@
 #include <time.h>
 
 #ifdef WITH_WEBSOCKETS
-#  include <libwebsockets.h>
+#  include "libwebsockets.h"
 #endif
 
 #include <mosquitto_broker.h>
--- a/src/mosquitto.c
+++ b/src/mosquitto.c
@@ -44,7 +44,7 @@
 #include <tcpd.h>
 #endif
 #ifdef WITH_WEBSOCKETS
-#  include <libwebsockets.h>
+#  include "libwebsockets.h"
 #endif
 
 #include <mosquitto_broker.h>
--- a/src/mosquitto_broker.h
+++ b/src/mosquitto_broker.h
@@ -21,7 +21,7 @@
 #include <stdio.h>
 
 #ifdef WITH_WEBSOCKETS
-#  include <libwebsockets.h>
+#  include "libwebsockets.h"
 
 #  if defined(LWS_LIBRARY_VERSION_NUMBER)
 #    define libwebsocket_callback_on_writable(A, B) lws_callback_on_writable((B))
--- a/src/read_handle_server.c
+++ b/src/read_handle_server.c
@@ -32,7 +32,7 @@
 #endif
 
 #ifdef WITH_WEBSOCKETS
-#  include <libwebsockets.h>
+#  include "libwebsockets.h"
 #endif
 
 #ifdef WITH_SYS_TREE
--- a/src/websockets.c
+++ b/src/websockets.c
@@ -29,7 +29,7 @@
 
 #ifdef WITH_WEBSOCKETS
 
-#include <libwebsockets.h>
+#include "libwebsockets.h"
 #include "mosquitto_internal.h"
 #include "mosquitto_broker.h"
 #include "mqtt3_protocol.h"
