--- a/src/mkheader.c
+++ b/src/mkheader.c
@@ -83,6 +83,9 @@
     {"i586-pc-kfreebsd-gnu"},
     {"x86_64-pc-linux-gnuhardened1", "x86_64-pc-linux-gnu" },
     {"powerpc-unknown-linux-gnuspe", "powerpc-unknown-linux-gnu" },
+    {"arm-openwrt-linux-gnu", "arm-unknown-linux-gnueabi"},
+    {"mipsel-openwrt-linux-gnu", "mipsel-unknown-linux-gnu"},
+    {"mips-openwrt-linux-gnu", "mips-unknown-linux-gnu"},
 
     { NULL }
   };
