# SER_RS485_RX_DURING_TX since Linux 3.2

--- a/lib/gensio_osops.c
+++ b/lib/gensio_osops.c
@@ -1687,8 +1687,10 @@ process_rs485(struct gensio_os_funcs *o,
 	    t->rs485.flags |= SER_RS485_RTS_ON_SEND;
 	} else if (strcmp(argv[i], "rts_after_send") == 0) {
 	    t->rs485.flags |= SER_RS485_RTS_AFTER_SEND;
+#ifdef SER_RS485_RX_DURING_TX
 	} else if (strcmp(argv[i], "rx_during_tx") == 0) {
 	    t->rs485.flags |= SER_RS485_RX_DURING_TX;
+#endif
 #ifdef SER_RS485_TERMINATE_BUS
 	} else if (strcmp(argv[i], "terminate_bus") == 0) {
 	    t->rs485.flags |= SER_RS485_TERMINATE_BUS;
