--- a/keepalived/vrrp/vrrp.c
+++ b/keepalived/vrrp/vrrp.c
@@ -2712,10 +2712,12 @@ open_vrrp_read_socket(sa_family_t family
 		if (family == AF_INET6) {
 			if (setsockopt(fd, IPPROTO_IPV6, IPV6_TRANSPARENT, &on, sizeof on))
 				log_message(LOG_INFO, "IPV6_TRANSPARENT failed %d - %m", errno);
-		} else
-#endif
+		} else if (setsockopt(fd, IPPROTO_IP, IP_FREEBIND, &on, sizeof on))
+			log_message(LOG_INFO, "IP_FREEBIND failed %d - %m", errno);
+#else
 		if (setsockopt(fd, family == AF_INET ? IPPROTO_IP : IPPROTO_IPV6, family == AF_INET ? IP_FREEBIND : IPV6_FREEBIND, &on, sizeof on))
 			log_message(LOG_INFO, "IP%s_FREEBIND failed %d - %m", family == AF_INET ? "" : "V6", errno);
+#endif
 
 		/* Bind to the local unicast address */
 		if (bind(fd, PTR_CAST_CONST(struct sockaddr, unicast_src), unicast_src->ss_family == AF_INET ? sizeof(struct sockaddr_in) : sizeof(struct sockaddr_in6))) {
