--- a/lib/isc/netmgr/udp.c
+++ b/lib/isc/netmgr/udp.c
@@ -30,6 +30,17 @@
 #include "netmgr-int.h"
 #include "uv-compat.h"
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* SO_REUSEPORT since Linux 3.9 */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,9,0)
+#define SO_REUSEPORT SO_REUSEADDR
+/* or #define SO_REUSEPORT 0x0200 */
+/* Allow local address and port reuse. */
+#endif
+
 static isc_result_t
 udp_send_direct(isc_nmsocket_t *sock, isc__nm_uvreq_t *req,
 		isc_sockaddr_t *peer);
