--- a/uvcdynctrl/udev/rules/80-uvcdynctrl.rules
+++ b/uvcdynctrl/udev/rules/80-uvcdynctrl.rules
@@ -6,4 +6,4 @@
 # udev versions confirmed to work: 141 (Ubuntu 9.04), 142 (Ubuntu 9.10)
 # udev versions confirmed not to work: 117 (Ubuntu 8.04)
 ###################################################################################################
-ACTION=="add", SUBSYSTEM=="video4linux", DRIVERS=="uvcvideo", RUN+="/lib/udev/uvcdynctrl"
+ACTION=="add", SUBSYSTEM=="video4linux", DRIVERS=="uvcvideo", RUN+="/opt/lib/udev/uvcdynctrl"
--- a/uvcdynctrl/udev/scripts/uvcdynctrl
+++ b/uvcdynctrl/udev/scripts/uvcdynctrl
@@ -19,7 +19,7 @@ uvcdynctrlpath=uvcdynctrl
 
 # Determine log destination
 if [ "$debug" != 0 ]; then
-	logfile=/var/log/uvcdynctrl-udev.log
+	logfile=/opt/var/log/uvcdynctrl-udev.log
 else
 	logfile=/dev/null
 fi
--- a/uvcdynctrl/CMakeLists.txt
+++ b/uvcdynctrl/CMakeLists.txt
@@ -102,11 +102,11 @@ install (
 # Note that having absolute paths here requires CPACK_SET_DESTDIR to be set to "ON".
 install (
 	FILES	udev/rules/80-uvcdynctrl.rules
-	DESTINATION /lib/udev/rules.d
+	DESTINATION /opt/lib/udev/rules.d
 )
 install (
 	PROGRAMS	udev/scripts/uvcdynctrl
-	DESTINATION	/lib/udev
+	DESTINATION	/opt/lib/udev
 )
 install (
 	DIRECTORY	data
