--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 if USE_SYSTEM_SHARED_LIB
 SUBDIRS = libcork libipset src
 else
-SUBDIRS = libsodium libcork libipset libudns libev src
+SUBDIRS = libcork libipset libudns src
 endif
 
 if ENABLE_DOCUMENTATION
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,9 +3,7 @@
 AM_CFLAGS = -g -O2 -Wall -Werror -Wno-deprecated-declarations -fno-strict-aliasing -std=gnu99 -D_GNU_SOURCE
 AM_CFLAGS += $(PTHREAD_CFLAGS)
 if !USE_SYSTEM_SHARED_LIB
-AM_CFLAGS += -I$(top_srcdir)/libev
 AM_CFLAGS += -I$(top_srcdir)/libudns
-AM_CFLAGS += -I$(top_srcdir)/libsodium/src/libsodium/include
 endif
 AM_CFLAGS += -I$(top_srcdir)/libipset/include
 AM_CFLAGS += -I$(top_srcdir)/libcork/include
@@ -14,12 +12,7 @@
 SS_COMMON_LIBS = $(top_builddir)/libipset/libipset.la \
                  $(top_builddir)/libcork/libcork.la \
                  $(INET_NTOP_LIB) $(LIBPCRE_LIBS)
-if USE_SYSTEM_SHARED_LIB
 SS_COMMON_LIBS += -lev -lsodium -lm
-else
-SS_COMMON_LIBS += $(top_builddir)/libev/libev.la \
-                  $(top_builddir)/libsodium/src/libsodium/libsodium.la
-endif
 
 bin_PROGRAMS = ss-local ss-tunnel
 if !BUILD_WINCOMPAT
