--- a/daemon/main.c
+++ b/daemon/main.c
@@ -604,8 +604,8 @@ void set_global_environment() {
 
     // set the path we need
     char path[1024 + 1], *p = getenv("PATH");
-    if(!p) p = "/bin:/usr/bin";
-    snprintfz(path, 1024, "%s:%s", p, "/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin");
+    if(!p) p = "/opt/bin:/opt/usr/bin:/bin:/usr/bin";
+    snprintfz(path, 1024, "%s:%s", p, "/opt/sbin:/opt/usr/sbin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin");
     setenv("PATH", config_get(CONFIG_SECTION_PLUGINS, "PATH environment variable", path), 1);
 
     // python options
