--- a/include/sound/uapi/asound.h
+++ b/include/sound/uapi/asound.h
@@ -903,13 +903,13 @@ struct snd_timer_status {
 #define SNDRV_TIMER_IOCTL_PAUSE		_IO('T', 0xa3)
 #define SNDRV_TIMER_IOCTL_TREAD64	_IOW('T', 0xa4, int)
 
-#if __BITS_PER_LONG == 64
+//#if __BITS_PER_LONG == 64
 #define SNDRV_TIMER_IOCTL_TREAD SNDRV_TIMER_IOCTL_TREAD_OLD
-#else
-#define SNDRV_TIMER_IOCTL_TREAD ((sizeof(__kernel_long_t) >= sizeof(time_t)) ? \
-				 SNDRV_TIMER_IOCTL_TREAD_OLD : \
-				 SNDRV_TIMER_IOCTL_TREAD64)
-#endif
+//#else
+//#define SNDRV_TIMER_IOCTL_TREAD ((sizeof(__kernel_long_t) >= sizeof(time_t)) ? \
+//				 SNDRV_TIMER_IOCTL_TREAD_OLD : \
+//				 SNDRV_TIMER_IOCTL_TREAD64)
+//#endif
 
 struct snd_timer_read {
 	unsigned int resolution;
