--- a/wsd.c
+++ b/wsd.c
@@ -59,7 +59,7 @@ static char wsd_sequence[UUIDLEN], wsd_e
 
 static void set_seed(void)
 {
-	FILE *fp = fopen("/etc/machine-id", "r");
+	FILE *fp = fopen("/opt/etc/machine-id", "r");
 	unsigned long seed;
 
 	time((time_t *)&seed);
@@ -94,7 +94,7 @@ static void uuid_random(char *uuid, size
 
 static void uuid_endpoint(char *uuid, size_t len)
 {
-	FILE *fp = fopen("/etc/machine-id", "r");
+	FILE *fp = fopen("/opt/etc/machine-id", "r");
 	int c, i = 0;
 	
 	if (!fp) {
@@ -102,7 +102,7 @@ static void uuid_endpoint(char *uuid, si
 	}
 
 	if (!fp) {
-		DEBUG(0, W, "Can't open required '/etc/machine-id' or '/proc/sys/kernel/random/boot_id'");
+		DEBUG(0, W, "Can't open required '/opt/etc/machine-id' or '/proc/sys/kernel/random/boot_id'");
 		return;
 	}
 
