--- a/wsd.c
+++ b/wsd.c
@@ -59,7 +59,7 @@
 
 static void set_seed(void)
 {
-	FILE *fp = fopen("/etc/machine-id", "r");
+	FILE *fp = fopen("/opt/etc/machine-id", "r");
 	unsigned long seed;
 
 	time((time_t *)&seed);
@@ -94,11 +94,11 @@
 
 static void uuid_endpoint(char *uuid, size_t len)
 {
-	FILE *fp = fopen("/etc/machine-id", "r");
+	FILE *fp = fopen("/opt/etc/machine-id", "r");
 	int c, i = 0;
 
 	if (!fp) {
-		DEBUG(0, W, "Can't open '/etc/machine-id', trying '/proc/sys/kernel/random/boot_id'");
+		DEBUG(0, W, "Can't open '/opt/etc/machine-id', trying '/proc/sys/kernel/random/boot_id'");
 		fp = fopen("/proc/sys/kernel/random/boot_id", "r");
 	}
 		
