--- a/make-linux.mk
+++ b/make-linux.mk
@@ -19,15 +19,6 @@
 #   redhat: build RPM packages; rpm dev tools must be present
 #
 
-# Automagically pick clang or gcc, with preference for clang
-# This is only done if we have not overridden these with an environment or CLI variable
-ifeq ($(origin CC),default)
-	CC=$(shell if [ -e /usr/bin/clang ]; then echo clang; else echo gcc; fi)
-endif
-ifeq ($(origin CXX),default)
-	CXX=$(shell if [ -e /usr/bin/clang++ ]; then echo clang++; else echo g++; fi)
-endif
-
 #UNAME_M=$(shell $(CC) -dumpmachine | cut -d '-' -f 1)
 
 INCLUDES?=
@@ -84,7 +75,7 @@
 
 ifeq ($(ZT_ENABLE_NETWORK_CONTROLLER),1)
 	DEFS+=-DZT_ENABLE_NETWORK_CONTROLLER
-	LDLIBS+=-L/usr/local/lib -lsqlite3
+	LDLIBS+=-lsqlite3
 	OBJS+=controller/SqliteNetworkController.o
 endif
 
@@ -110,7 +101,7 @@
 	override CFLAGS+=-Wall -fPIE -pthread $(INCLUDES) -DNDEBUG $(DEFS)
 	CXXFLAGS?=-O3 -fstack-protector-strong
 	override CXXFLAGS+=-Wall -Wno-unused-result -Wreorder -fPIE -fno-rtti -pthread $(INCLUDES) -DNDEBUG $(DEFS)
-	LDFLAGS=-pie -Wl,-z,relro,-z,now
+	LDFLAGS+=-Wl,-z,relro,-z,now
 	STRIP?=strip
 	STRIP+=--strip-all
 endif
@@ -121,7 +112,7 @@
 #LDFLAGS=
 #STRIP=echo
 
-all:	one manpages
+all:	one
 
 one:	$(OBJS) service/OneService.o one.o osdep/LinuxEthernetTap.o
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o zerotier-one $(OBJS) service/OneService.o one.o osdep/LinuxEthernetTap.o $(LDLIBS)
