--- a/pkg_resources/_vendor/appdirs.py
+++ b/pkg_resources/_vendor/appdirs.py
@@ -145,7 +145,7 @@
         # XDG default for $XDG_DATA_DIRS
         # only first, if multipath is False
         path = os.getenv('XDG_DATA_DIRS',
-                         os.pathsep.join(['/usr/local/share', '/usr/share']))
+                         os.pathsep.join(['/opt/share', '/opt/usr/share', '/usr/local/share', '/usr/share']))
         pathlist = [os.path.expanduser(x.rstrip(os.sep)) for x in path.split(os.pathsep)]
         if appname:
             if version:
