--- a/scripts/sendsms
+++ b/scripts/sendsms
@@ -1,4 +1,4 @@
-#!/bin/ash
+#!/bin/sh
 # This script send a text sms at the command line by creating
 # a sms file in the outgoing queue.
 
@@ -18,7 +18,7 @@ KEYS=""
 # When creating keys, remember to use -n for echo:
 # echo -n "key" | md5sum
 
-smsd_group="smsd"
+smsd_group="root"
 
 # Will need echo which accepts -n argument:
 ECHO=echo
@@ -96,7 +96,7 @@ for destination in $destinations
 do
   echo "To: $destination"
 
-  TMPFILE=`mktemp /tmp/smsd_XXXXXX`
+  TMPFILE=`mktemp /opt/tmp/smsd_XXXXXX`
 
   $ECHO "To: $destination" >> $TMPFILE
   [ -n "$ALPHABET" ] && $ECHO "$ALPHABET" >> $TMPFILE
@@ -113,6 +113,6 @@ do
 
   chmod 0660 $TMPFILE
 
-  FILE=`mktemp /var/spool/sms/outgoing/send_XXXXXX`
+  FILE=`mktemp /opt/var/spool/sms/outgoing/send_XXXXXX`
   mv $TMPFILE $FILE
 done
--- a/scripts/sms2html
+++ b/scripts/sms2html
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # This script converts a received sms file into a html file. 
 
--- a/scripts/sms2unicode
+++ b/scripts/sms2unicode
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # This script converts a received sms file into a pure unicode text file.
 
--- a/scripts/unicode2sms
+++ b/scripts/unicode2sms
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # This script converts a pure unicode text file into an sms file for sending.
 
--- a/examples/smsd.conf.easy
+++ b/examples/smsd.conf.easy
@@ -1,7 +1,7 @@
 # Example smsd.conf. Read the manual for a description
 
 devices = GSM1
-logfile = /var/log/smsd.log
+logfile = /opt/var/log/smsd.log
 loglevel = 7
 
 [GSM1]
