--- a/src/smsd_cfg.c	2012-04-12 13:12:33.195184453 +0400
+++ b/src/smsd_cfg.c	2012-04-12 13:22:07.012336003 +0400
@@ -312,14 +312,14 @@
   log_unmodified = 0;
   alarmlevel=LOG_WARNING;
 
-  strcpy(d_spool,"/var/spool/sms/outgoing");
-  strcpy(d_incoming,"/var/spool/sms/incoming");
+  strcpy(d_spool,"/opt/var/spool/sms/outgoing");
+  strcpy(d_incoming,"/opt/var/spool/sms/incoming");
   *d_incoming_copy = 0; // 3.1.16beta2.
   *d_report = 0;
   *d_report_copy = 0; // 3.1.17.
   *d_phonecalls = 0;
   *d_saved = 0;
-  strcpy(d_checked,"/var/spool/sms/checked");
+  strcpy(d_checked,"/opt/var/spool/sms/checked");
   d_failed[0]=0;
   d_failed_copy[0]=0; // 3.1.17.
   d_sent[0]=0;
@@ -393,8 +393,8 @@
   username[0] = 0;
   groupname[0] = 0;
 
-  strcpy(infofile, "/var/run/smsd.working");
-  strcpy(pidfile, "/var/run/smsd.pid");
+  strcpy(infofile, "/opt/var/run/smsd.working");
+  strcpy(pidfile, "/opt/var/run/smsd.pid");
 
   terminal = 0;
   os_cygwin = 0;
--- a/src/smsd.c	2010-09-21 13:47:47.000000000 +0400
+++ b/src/smsd.c	2012-04-12 13:22:49.161297293 +0400
@@ -7666,7 +7666,7 @@
     if (p && *p && strlen(p) < sizeof(tmpdir))
       snprintf(tmpdir, sizeof(tmpdir), "%s", p);
     else
-      strcpy(tmpdir, "/tmp");
+      strcpy(tmpdir, "/opt/tmp");
   }
 
   break_workless_delay = 0;
