--- a/config.c
+++ b/config.c
@@ -343,7 +343,7 @@ static const char *config_def_dpd_idle(v
 
 static const char *config_ca_dir(void)
 {
-	return "/etc/ssl/certs";
+	return "/opt/etc/ssl/certs";
 }
 
 static const char *config_def_auth_mode(void)
@@ -363,7 +363,7 @@ static const char *config_def_app_versio
 
 static const char *config_def_script(void)
 {
-	return "/etc/vpnc/vpnc-script";
+	return "/opt/etc/vpnc/vpnc-script";
 }
 
 static const char *config_def_pid_file(void)
@@ -659,7 +659,7 @@ static char *get_config_filename(const c
 {
 	char *realname;
 
-	asprintf(&realname, "%s%s%s", index(name, '/') ? "" : "/etc/vpnc/", name, add_dot_conf ? ".conf" : "");
+	asprintf(&realname, "%s%s%s", index(name, '/') ? "" : "/opt/etc/vpnc/", name, add_dot_conf ? ".conf" : "");
 	return realname;
 }
 
@@ -895,8 +895,8 @@ void do_config(int argc, char **argv)
 	}
 
 	if (!got_conffile) {
-		read_config_file("/etc/vpnc/default.conf", config, 1);
-		read_config_file("/etc/vpnc.conf", config, 1);
+		read_config_file("/opt/etc/vpnc/default.conf", config, 1);
+		read_config_file("/opt/etc/vpnc.conf", config, 1);
 	}
 
 	if (!print_config) {
