--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-prefix = /usr
+prefix = /opt
 
 ifdef DESTDIR
 BASEDIR := $(DESTDIR)
@@ -77,9 +77,6 @@ all: Makefile.depend whois pos
 whois: $(whois_OBJECTS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(whois_LDADD) $(LIBS)
 
-mkpasswd: $(mkpasswd_OBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(mkpasswd_LDADD) $(LIBS)
-
 ##############################################################################
 version.h: debian/changelog make_version_h.pl
 	$(PERL) make_version_h.pl $< > $@
@@ -117,7 +114,7 @@ afl-run:
 	nice afl-fuzz -i ../afl_in -o ../afl_out -- ./whois
 
 ##############################################################################
-install: install-whois install-mkpasswd install-pos
+install: install-whois install-pos
 
 install-whois: whois
 	$(INSTALL) -d $(BASEDIR)$(prefix)/bin/
@@ -149,6 +146,7 @@ pos:
 
 depend: Makefile.depend
 Makefile.depend:
+	rm -f mkpasswd.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -MM -MG *.c > $@
 
 -include Makefile.depend
--- a/po/Makefile
+++ b/po/Makefile
@@ -1,4 +1,4 @@
-prefix = /usr
+prefix = /opt
 
 ifdef DESTDIR
 BASEDIR := $(DESTDIR)
@@ -12,7 +12,7 @@ PACKAGE = whois
 
 CATALOGS = cs.mo da.mo de.mo el.mo es.mo eu.mo fi.mo fr.mo it.mo ja.mo pl.mo pt_BR.mo ru.mo zh_CN.mo
 
-POTFILES=../whois.c ../mkpasswd.c
+POTFILES=../whois.c
 
 all: $(PACKAGE).pot $(CATALOGS)
 
