--- a/Makefile
+++ b/Makefile
@@ -7,20 +7,20 @@ SPECFILE	:= keyutils.spec
 NO_GLIBC_KEYERR	:= 0
 NO_ARLIB	:= 0
 NO_SOLIB	:= 0
-ETCDIR		:= /etc
-BINDIR		:= /bin
-SBINDIR		:= /sbin
-SHAREDIR	:= /usr/share/keyutils
-MANDIR		:= /usr/share/man
+ETCDIR		:= /opt/etc
+BINDIR		:= /opt/bin
+SBINDIR		:= /opt/sbin
+SHAREDIR	:= /opt/share/keyutils
+MANDIR		:= /opt/share/man
 MAN1		:= $(MANDIR)/man1
 MAN3		:= $(MANDIR)/man3
 MAN5		:= $(MANDIR)/man5
 MAN7		:= $(MANDIR)/man7
 MAN8		:= $(MANDIR)/man8
-INCLUDEDIR	:= /usr/include
+INCLUDEDIR	:= /opt/include
 LN		:= ln
 LNS		:= $(LN) -sf
-PREFIX 		:= /usr
+PREFIX 		:= /opt
 
 ###############################################################################
 #
@@ -68,12 +68,10 @@ BUILDFOR	:= $(shell file /usr/bin/make |
 
 ifeq ($(origin CFLAGS),undefined)
 ifeq ($(BUILDFOR),32-bit)
-CFLAGS		+= -m32
 LIBDIR		:= /lib
 USRLIBDIR	:= /usr/lib
 else
 ifeq ($(BUILDFOR),64-bit)
-CFLAGS		+= -m64
 LIBDIR		:= /lib64
 USRLIBDIR	:= /usr/lib64
 endif
@@ -200,7 +198,7 @@ ifeq ($(NO_SOLIB),0)
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
 	sed \
 	-e 's,@VERSION\@,$(VERSION),g' \
 	-e 's,@prefix\@,$(PREFIX),g' \
--- a/key.dns_resolver.c
+++ b/key.dns_resolver.c
@@ -46,7 +46,7 @@ static const char key_type[] = "dns_reso
 static const char a_query_type[] = "a";
 static const char aaaa_query_type[] = "aaaa";
 static const char afsdb_query_type[] = "afsdb";
-static const char *config_file = "/etc/keyutils/key.dns_resolver.conf";
+static const char *config_file = "/opt/etc/keyutils/key.dns_resolver.conf";
 static bool config_specified = false;
 key_serial_t key;
 static int verbose;
--- a/request-key.c
+++ b/request-key.c
@@ -278,8 +278,8 @@ inaccessible:
 static void lookup_action(struct parameters *params)
 {
 	if (!xlocaldirs) {
-		scan_conf_dir(params, "/etc/request-key.d");
-		scan_conf_file(params, AT_FDCWD, "/etc/request-key.conf");
+		scan_conf_dir(params, "/opt/etc/request-key.d");
+		scan_conf_file(params, AT_FDCWD, "/opt/etc/request-key.conf");
 	} else {
 		scan_conf_dir(params, "request-key.d");
 		scan_conf_file(params, AT_FDCWD, "request-key.conf");
