--- a/Makefile
+++ b/Makefile
@@ -6,20 +6,20 @@ SPECFILE	:= keyutils.spec
 NO_GLIBC_KEYERR	:= 0
 NO_ARLIB	:= 0
 NO_SOLIB	:= 0
-ETCDIR		:= /etc
-BINDIR		:= /bin
-SBINDIR		:= /sbin
-SHAREDIR	:= /usr/share/keyutils
-MANDIR		:= /usr/share/man
+ETCDIR		:= /opt/etc
+BINDIR		:= /opt/bin
+SBINDIR		:= /opt/sbin
+SHAREDIR	:= /opt/share/keyutils
+MANDIR		:= /opt/share/man
 MAN1		:= $(MANDIR)/man1
 MAN3		:= $(MANDIR)/man3
 MAN5		:= $(MANDIR)/man5
 MAN7		:= $(MANDIR)/man7
 MAN8		:= $(MANDIR)/man8
-INCLUDEDIR	:= /usr/include
+INCLUDEDIR	:= /opt/include
 LN		:= ln
 LNS		:= $(LN) -sf
-PREFIX 		:= /usr
+PREFIX 		:= /opt
 
 ###############################################################################
 #
@@ -67,12 +67,10 @@ BUILDFOR	:= $(shell file /usr/bin/make |
 
 ifeq ($(origin CFLAGS),undefined)
 ifeq ($(BUILDFOR),32-bit)
-CFLAGS		+= -m32
 LIBDIR		:= /lib
 USRLIBDIR	:= /usr/lib
 else
 ifeq ($(BUILDFOR),64-bit)
-CFLAGS		+= -m64
 LIBDIR		:= /lib64
 USRLIBDIR	:= /usr/lib64
 endif
@@ -184,7 +182,7 @@ ifeq ($(NO_SOLIB),0)
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
 	sed \
 	-e 's,@VERSION\@,$(VERSION),g' \
 	-e 's,@prefix\@,$(PREFIX),g' \
