--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -172,7 +172,7 @@ elif os.name == "posix":
             null = open(os.devnull, 'wb')
             try:
                 with null:
-                    proc = subprocess.Popen(('/sbin/ldconfig', '-r'),
+                    proc = subprocess.Popen(('/opt/sbin/ldconfig', '-r'),
                                             stdout=subprocess.PIPE,
                                             stderr=null)
             except OSError:  # E.g. command not found
@@ -258,7 +258,7 @@ elif os.name == "posix":
             null = open(os.devnull, 'wb')
             try:
                 with null:
-                    p = subprocess.Popen(['/sbin/ldconfig', '-p'],
+                    p = subprocess.Popen(['/opt/sbin/ldconfig', '-p'],
                                           stderr=null,
                                           stdout=subprocess.PIPE,
                                           env=env)
