--- a/configure
+++ b/configure
@@ -15202,9 +15202,9 @@
   $as_echo_n "(cached) " >&6
 else
   if test "$cross_compiling" = yes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: cannot check job control if cross-compiling -- defaulting to missing" >&5
-$as_echo "$as_me: WARNING: cannot check job control if cross-compiling -- defaulting to missing" >&2;}
-     bash_cv_job_control_missing=missing
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: cannot check job control if cross-compiling -- defaulting to presentg" >&5
+$as_echo "$as_me: WARNING: cannot check job control if cross-compiling -- defaulting to present" >&2;}
+     bash_cv_job_control_missing=present
 
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
