--- a/bandwidthd.c
+++ b/bandwidthd.c
@@ -210,20 +210,20 @@ void makepidfile(pid_t pid)
 	{
 	FILE *pidfile;
 
-	pidfile = fopen("/var/run/bandwidthd.pid", "wt");
+	pidfile = fopen("/opt/var/run/bandwidthd.pid", "wt");
 	if (pidfile) 
 		{
 		if (fprintf(pidfile, "%d\n", pid) == 0) 
 			{
-			syslog(LOG_ERR, "Bandwidthd: failed to write '%d' to /var/run/bandwidthd.pid", pid);
+			syslog(LOG_ERR, "Bandwidthd: failed to write '%d' to /opt/var/run/bandwidthd.pid", pid);
 			fclose(pidfile);
-			unlink("/var/run/bandwidthd.pid");
+			unlink("/opt/var/run/bandwidthd.pid");
 			}
 		else
 			fclose(pidfile);		
 		}
 	else
-		syslog(LOG_ERR, "Could not open /var/run/bandwidthd.pid for write");
+		syslog(LOG_ERR, "Could not open /opt/var/run/bandwidthd.pid for write");
 	}
 
 
@@ -258,13 +258,13 @@ int main(int argc, char **argv)
 
 	openlog("bandwidthd", LOG_CONS, LOG_DAEMON);
 
-	if (stat("/etc/bandwidthd.conf", &StatBuf))
+	if (stat("/opt/etc/config/bandwidthd.conf", &StatBuf))
 		{
 		chdir(INSTALL_DIR);
 		if (stat("./etc/bandwidthd.conf", &StatBuf))
 			{
-			printf("Cannot find /etc/bandwidthd.conf or %s/etc/bandwidthd.conf\n", INSTALL_DIR);
-			syslog(LOG_ERR, "Cannot find /etc/bandwidthd.conf or %s/etc/bandwidthd.conf", INSTALL_DIR);
+			printf("Cannot find /opt/etc/config/bandwidthd.conf\n", INSTALL_DIR);
+			syslog(LOG_ERR, "Cannot find /opt/etc/config/bandwidthd.conf", INSTALL_DIR);
 			exit(1);
 			}
 		else
@@ -274,7 +274,7 @@ int main(int argc, char **argv)
 		}
 	else
 	        {
-			bdconfig_in = fopen("/etc/bandwidthd.conf", "rt");
+			bdconfig_in = fopen("/opt/etc/config/bandwidthd.conf", "rt");
 		}
 		
 	if (!bdconfig_in)
@@ -349,10 +349,10 @@ int main(int argc, char **argv)
 
 	if (config.graph)
 		{
-		bd_CollectingData("htdocs/index.html");
-		bd_CollectingData("htdocs/index2.html");
-		bd_CollectingData("htdocs/index3.html");
-		bd_CollectingData("htdocs/index4.html");
+		bd_CollectingData("/opt/share/www/bandwidthd/index.html");
+		bd_CollectingData("/opt/share/www/bandwidthd/index2.html");
+		bd_CollectingData("/opt/share/www/bandwidthd/index3.html");
+		bd_CollectingData("/opt/share/www/bandwidthd/index4.html");
 		}
 
 	/* detach from console. */
--- a/bandwidthd.service
+++ b/bandwidthd.service
@@ -3,7 +3,7 @@ Description=Bandwidthd Network Traffic M
 After=network.target
 
 [Service]
-ExecStart=/usr/sbin/bandwidthd -D
+ExecStart=/opt/sbin/bandwidthd -D
 
 [Install]
 WantedBy=multi-user.target
--- a/httpd.conf
+++ b/httpd.conf
@@ -1,8 +1,8 @@
 #
 # bandwidthd configuration
 #
-Alias /bandwidthd /usr/share/bandwidthd
-<Directory /usr/share/bandwidthd>
+Alias /bandwidthd /opt/share/www/bandwidthd
+<Directory /opt/share/www/bandwidthd>
   php_value short_open_tag on
   Require ip 127.0.0.1
 </Directory>
\ No newline at end of file
--- a/configure
+++ b/configure
@@ -80,8 +80,8 @@ exec 6>&1
 #
 # Initializations.
 #
-ac_default_prefix=/usr/local
-cross_compiling=no
+ac_default_prefix=/opt
+cross_compiling=yes
 subdirs=
 MFLAGS= MAKEFLAGS=
 SHELL=${CONFIG_SHELL-/bin/sh}
@@ -138,10 +138,10 @@ ac_init_version=false
 # The variables have the same names as the options, with
 # dashes changed to underlines.
 cache_file=/dev/null
-exec_prefix=NONE
+exec_prefix=/opt
 no_create=
 no_recursion=
-prefix=NONE
+prefix=/opt
 program_prefix=NONE
 program_suffix=NONE
 program_transform_name=s,x,x,
