--- a/etc/bandwidthd.conf
+++ b/etc/bandwidthd.conf
@@ -52,7 +52,7 @@ graph false
 #meta_refresh 150
 
 # SQLite DB path
-sqlite_filename "/var/www/bandwidthd/stats.db"
+sqlite_filename "/opt/var/bandwidthd/stats.db"
 
 # Provide a default sensor name to avoid "unset" value
 sensor_id "default"
--- a/phphtdocs/config.conf.php
+++ b/phphtdocs/config.conf.php
@@ -1,6 +1,6 @@
-<?
+<?php
 define("DFLT_WIDTH", 900);
 define("DFLT_HEIGHT", 256);
 define("DFLT_INTERVAL", INT_DAILY);
 
-$db_connect_string = "sqlite:/var/www/bandwidthd/stats.db";
+$db_connect_string = "sqlite:/opt/var/bandwidthd/stats.db";
