--- a/contrib/i2pd.conf
+++ b/contrib/i2pd.conf
@@ -8,19 +8,19 @@
 
 ## Tunnels config file
 ## Default: ~/.i2pd/tunnels.conf or /var/lib/i2pd/tunnels.conf
-tunconf = /etc/i2pd/tunnels.conf
+tunconf = /opt/etc/i2pd/tunnels.conf
 
 ## Tunnels config files path
 ## Use that path to store separated tunnels in different config files.
 ## Default: ~/.i2pd/tunnels.d or /var/lib/i2pd/tunnels.d
-tunnelsdir = /etc/i2pd/tunnels.d
+tunnelsdir = /opt/etc/i2pd/tunnels.d
 
 ## Path to certificates used for verifying .su3, families
 ## Default: ~/.i2pd/certificates or /var/lib/i2pd/certificates
-certsdir = /usr/share/i2pd/certificates
+certsdir = /opt/share/i2pd/certificates
 
 ## Where to write pidfile (default: i2pd.pid, not used in Windows)
-# pidfile = /run/i2pd.pid
+pidfile = /opt/var/run/i2pd.pid
 
 ## Logging configuration section
 ## By default logs go to stdout with level 'info' and higher
@@ -30,9 +30,9 @@ certsdir = /usr/share/i2pd/certificates
 ##  * stdout - print log entries to stdout
 ##  * file - log entries to a file
 ##  * syslog - use syslog, see man 3 syslog
-# log = file
+log = file
 ## Path to logfile (default - autodetect)
-# logfile = /var/log/i2pd/i2pd.log
+logfile = /opt/var/log/i2pd.log
 ## Log messages above this level (debug, info, *warn, error, none)
 ## If you set it to none, logging will be disabled
 loglevel = none
@@ -101,7 +101,7 @@ ipv6 = false
 ## Uncomment and set to 'false' to disable Web Console
 enabled = true
 ## Address and port service will listen on
-address = 192.168.1.1
+address = 127.0.0.1
 port = 7070
 ## Path to web console, default "/"
 # webroot = /
@@ -118,7 +118,7 @@ port = 7070
 ## Uncomment and set to 'false' to disable HTTP Proxy
 enabled = true
 ## Address and port service will listen on
-address = 192.168.1.1
+address = 127.0.0.1
 port = 4444
 ## Optional keys file for proxy local destination
 # keys = http-proxy-keys.dat
@@ -132,7 +132,7 @@ port = 4444
 ## Uncomment and set to 'false' to disable SOCKS Proxy
 enabled = true
 ## Address and port service will listen on
-address = 192.168.1.1
+address = 127.0.0.1
 port = 4447
 ## Optional keys file for proxy local destination
 # keys = socks-proxy-keys.dat
