--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -5,7 +5,7 @@
 Interface "eth0"
 
 # location of the database directory
-DatabaseDir "/var/lib/vnstat"
+DatabaseDir "/opt/var/lib/vnstat"
 
 # locale (LC_ALL) ("-" = use system locale)
 Locale "-"
@@ -87,13 +87,13 @@
 DaemonGroup ""
 
 # how often (in seconds) interface data is updated
-UpdateInterval 30
+UpdateInterval 60
 
 # how often (in seconds) interface status changes are checked
-PollInterval 5
+PollInterval 30
 
 # how often (in minutes) data is saved to file
-SaveInterval 5
+SaveInterval 30
 
 # how often (in minutes) data is saved when all interface are offline
 OfflineSaveInterval 30
@@ -115,10 +115,10 @@
 UpdateFileOwner 1
 
 # file used for logging if UseLogging is set to 1
-LogFile "/var/log/vnstat/vnstat.log"
+LogFile "/opt/var/log/vnstat.log"
 
 # file used as daemon pid / lock file
-PidFile "/var/run/vnstat/vnstat.pid"
+PidFile "/opt/var/run/vnstat.pid"
 
 
 # vnstati
