--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -5,7 +5,7 @@
 Interface "eth0"
 
 # location of the database directory
-DatabaseDir "/var/lib/vnstat"
+DatabaseDir "/opt/var/lib/vnstat"
 
 # locale (LC_ALL) ("-" = use system locale)
 Locale "-"
@@ -126,10 +126,10 @@ CreateDirs 1
 UpdateFileOwner 1
 
 # file used for logging if UseLogging is set to 1
-LogFile "/var/log/vnstat/vnstat.log"
+LogFile "/opt/var/log/vnstat.log"
 
 # file used as daemon pid / lock file
-PidFile "/var/run/vnstat/vnstat.pid"
+PidFile "/opt/var/run/vnstat.pid"
 
 
 # vnstati
