--- a/src/core/or/connection_edge.c
+++ b/src/core/or/connection_edge.c
@@ -130,6 +130,15 @@
 #include <linux/if.h>
 #endif
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* IP6T_SO_ORIGINAL_DST since Linux 3.8 */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
+#define IP6T_SO_ORIGINAL_DST	80
+#endif
+
 #ifdef HAVE_LINUX_NETFILTER_IPV6_IP6_TABLES_H
 #include <linux/netfilter_ipv6/ip6_tables.h>
 #if defined(IP6T_SO_ORIGINAL_DST)
--- a/src/lib/sandbox/sandbox.c
+++ b/src/lib/sandbox/sandbox.c
@@ -83,6 +83,15 @@
 #include <linux/netfilter_ipv6/ip6_tables.h>
 #endif
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* IP6T_SO_ORIGINAL_DST since Linux 3.8 */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
+#define IP6T_SO_ORIGINAL_DST	80
+#endif
+
 #if defined(HAVE_EXECINFO_H) && defined(HAVE_BACKTRACE) && \
   defined(HAVE_BACKTRACE_SYMBOLS_FD) && defined(HAVE_SIGACTION)
 #define USE_BACKTRACE
