--- a/configure.ac
+++ b/configure.ac
@@ -440,8 +440,8 @@ yes)
 		AC_DEFINE_UNQUOTED(FC_DEFAULT_FONTS, "WINDOWSFONTDIR", 
 				   [Windows font directory])
 	else
-		FC_DEFAULT_FONTS="/usr/share/fonts"
-		AC_DEFINE_UNQUOTED(FC_DEFAULT_FONTS, "/usr/share/fonts", 
+		FC_DEFAULT_FONTS="/opt/share/fonts"
+		AC_DEFINE_UNQUOTED(FC_DEFAULT_FONTS, "/opt/share/fonts", 
 				   [System font directory])
 	fi
 	;;
